{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Extract source SBML-qual model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import libsbml\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from collections import Counter\n",
    "import re\n",
    "\n",
    "biodivine_model_dir = '/Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/140_BREAST-CANCER-DRUG-RESISTANCE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/233_BREAST-CANCER-SIGNALLING-PATHWAYS-T47D-COMPLETE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/264_PREIMPLANTATION-EMBRYO-TROPHECTODERM-1\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/084_ABA-INDUCED-STOMATAL-CLOSURE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/237_COLORECTAL-TUMORIGENESIS-REVERSION-SWITCH\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/265_PREIMPLANTATION-EMBRYO-TROPHECTODERM-1-MATURE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/075_INFLAMMATORY-BOWEL-DISEASE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/229_CALCIUM-SIGNALING-STOMATAL-CLOSURE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/276_GLUCAGON-AND-INSULIN-PRODUCTION\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/199_HEPATOCELLULAR-CARCINOMA-COMPARTMENTALIZED\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/250_MYELOID-LEUKEMIA-PIM-PATHWAY\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/999_TEMPLATE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/260_IMMUNE-SURVEILLANCE-OF-PRIMARY-MELANOMA-COMBINED\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/245_CARDIOMYOCYTE-DIFFERENTIATION\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/211_EPITHELIAL-DERIVED-CANCER-CELLS\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/227_HEDGEHOG-SIGNALING-PATHWAY\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/270_PREECLAMPSIA-PATHOGENESIS\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/251_CELL-CYCLE-POOL\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/206_EPITHELIAL-MESENCHYMAL-TRANSITION-IN-BREAST\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/103_PANCREATIC-CANCER-MICROENVIRONMENT\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/279_EMT-HEDGEHOG-SIGNALING\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/247_MAMMALIAN-EPIDERMIS\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/201_ONKOGENE-ROLE-OF-INCRNA-ANRIL\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/094_MACROPHAGE-POLARIZATION-STATES\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/138_APOPTOSIS-UPDATED\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/262_CANCER-SIGNALLING-TUTORIAL\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/223_ANGIOGENESIS-MICROENVIRONMENT\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/253_STOMAL-OSMOREGULATION-IN-CANCER\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/107_DNA-DAMAGE-INDUCED-AUTOPHAGY\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/269_COLORECTAL-CANCER-GSE8671\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/271_BLOOD-STEM-CELL-HETEROGENEITY\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/266_PREIMPLANTATION-EMBRYO-TROPHECTODERM-2\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/234_BREAST-CANCER-SIGNALLING-PATHWAYS-MDAMB468-COMPLETE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/273_HSC-MAINTENANCE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/174_HEPATOCELLULAR-CARCINOMA-REDUCED\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/261_SARS-COV-2-HOST-INTERACTION\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/256_LACTIC-ACID-LYMPH-NODE-STROMA\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/254_MYC-HETEROGENEITY-IN-CANCER-IN-VIVO\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/255_MYC-HETEROGENEITY-IN-CANCER-IN-VITRO\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/232_BREAST-CANCER-PATHWAYS-MCF7-COMPLETE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/221_MYCOBACTERIAL-LATENCY\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/100_ACUTE-MYELOID-LEUKEMIA\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/222_TEMPORAL-PROTEIN-EXPRESSION\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/246_C-ELEGANS-VULVAL-DEVELOPMENT\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/278_STOCHASTICITY-IN-HEMATOPOIESIS\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/212_ESCHERICHIA-COLI-TRYPTOPHAN\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/220_HEALING-RESPONSE-IN-LEISHMANIASIS\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/263_METABOLISM-TUTORIAL\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/079_COLORECTAL-TUMORIGENESIS\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/228_INTEGRATION-OF-SIGNALING-AND-KINETIC-MODELS\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/244_LIMPHOPOIESIS\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/101_GUARD-CELL-CO2-SIGNALLING\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/105_PLANT-GUARD-CELL-SIGNALLING\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/225_INTEGRATED-PROSTATE-CANCER-NETWORK\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/087_INFLAMMATORY-GENE-EXPRESSION-IN-MACROPHAGES\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/258_IMMUNE-SURVEILLANCE-OF-PRIMARY-MELANOMA\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/173_GLUCOSE-REPRESSION-SIGNALLING\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/144_SNF1-AMPK-PATHWAY\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/259_IMMUNE-SURVEILLANCE-OF-PRIMARY-MELANOMA-LC\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/098_MACROPHAGE-POLARIZATION-EXTENDED\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/207_BREAST-CANCER-TUMOUR\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/177_MYELOID-PROGENITORS\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/226_B-CELL-APOPTOSIS\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/230_HEPATOCYTE-GROWTH-FACTOR\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/268_COLORECTAL-CANCER-GSE1323\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/257_HOXA9-CANCER-SWITCH\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/205_EPITHELIAL-MESENCHYMAL-TRANSITION-IN-BLADDER\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/102_PANCREATIC-CANCER-MICROENVIRONMENT-REDUCED\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/248_DIABETES-METABOLIC-DISTURBANCE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/210_DRUG-SYNERGY-PREDICTION\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/208_HEMATOPOIESIS-AGING\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/267_PREIMPLANTATION-EMBRYO-TROPHECTODERM-2-MATURE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/224_BORTEZOMIB-PHARMACODYNAMIC-HETEROGENEITY\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/235_BREAST-CANCER-SIGNALLING-PATHWAYS-MDAMB231-COMPLETE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/252_MAMMALIAN-EPIDERMIS-2D\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/249_MYELOID-LEUKEMIA-DRUG-TARGETS\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/231_BREAST-CANCER-SIGNALLING-PATHWAYS\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/092_HEPATOCELLULAR-CARCINOMA\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/236_BREAST-CANCER-SIGNALLING-PATHWAYS-BT549-COMPLETE\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/202_ONKOGENE-ROLE-OF-INCRNA-XIST\n",
      "'source.sbml' not found in: /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/sources/281_EMT-SWITCH\n"
     ]
    }
   ],
   "source": [
    "# Extract source files from BioDivine\n",
    "import os\n",
    "import shutil\n",
    "\n",
    "source_directory = biodivine_model_dir + \"sources\"\n",
    "destination_directory = biodivine_model_dir\n",
    "\n",
    "# Ensure the destination directory exists\n",
    "os.makedirs(destination_directory, exist_ok=True)\n",
    "# Iterate through the subfolders in the source directory\n",
    "for subfolder in os.listdir(source_directory):\n",
    "    subfolder_path = os.path.join(source_directory, subfolder)\n",
    "    # Check if it's a directory\n",
    "    if os.path.isdir(subfolder_path):\n",
    "        # Path to the 'source.sbml' file in the current subfolder\n",
    "        sbml_file_path = os.path.join(subfolder_path, \"source.sbml\")\n",
    "        # Check if the file exists\n",
    "        if os.path.isfile(sbml_file_path):\n",
    "            # Define the target path in the destination folder\n",
    "            target_path = os.path.join(destination_directory, f\"{subfolder}_source.sbml\")\n",
    "            # Copy the file to the destination folder\n",
    "            shutil.copy(sbml_file_path, target_path)\n",
    "            # print(f\"Copied: {sbml_file_path} to {target_path}\")\n",
    "        else:\n",
    "            print(f\"'source.sbml' not found in: {subfolder_path}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotation statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "RDF_TAG_ITEM = ['rdf:RDF',\n",
    "                'xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\"',\n",
    "                'xmlns:dcterms=\"http://purl.org/dc/terms/\"',\n",
    "                'xmlns:vcard4=\"http://www.w3.org/2006/vcard/ns#\"',\n",
    "                'xmlns:bqbiol=\"http://biomodels.net/biology-qualifiers/\"',\n",
    "                'xmlns:bqmodel=\"http://biomodels.net/model-qualifiers/\"']\n",
    "RDF_TAG = ' '.join(RDF_TAG_ITEM)\n",
    "\n",
    "def get_gene_annotation_string(sbml_file):\n",
    "    \"\"\"\n",
    "    Extract GeneProducts annotatino string from an SBML-qual model.\n",
    "\n",
    "    Parameters:\n",
    "    - sbml_file (str): Path to the SBML file.\n",
    "\n",
    "    Returns:\n",
    "    - list: List of GeneProduct annotation strings.\n",
    "    \"\"\"\n",
    "    # Read the SBML model\n",
    "    reader = libsbml.SBMLReader()\n",
    "    document = reader.readSBML(sbml_file)\n",
    "    model = document.getModel()\n",
    "    if model is None:\n",
    "        print(f\"Failed to read the model: {sbml_file}\")\n",
    "        return []\n",
    "    # Check if the model uses the qual package\n",
    "    qual_plugin = model.getPlugin(\"qual\")\n",
    "    if not qual_plugin:\n",
    "        print(f\"The model does not use the qual package: {sbml_file}\")\n",
    "        return []\n",
    "    # Get GeneProducts from the plugin\n",
    "    gene_annotations = []\n",
    "    species = qual_plugin.getListOfQualitativeSpecies()\n",
    "    for i in range(len(species)):\n",
    "        one_species = species[i]\n",
    "        annotation_string = one_species.getAnnotationString()\n",
    "        gene_annotations.append(annotation_string)\n",
    "    return gene_annotations\n",
    "\n",
    "def extract_ontology_from_items(items_list):\n",
    "    \"\"\"\n",
    "    Extract ontology from items and return a flat list of tuples.\n",
    "    Each tuple contains (ontology type, ontology id).\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    items_list : list\n",
    "        A list of string items containing ontology annotations.\n",
    "\n",
    "    Returns\n",
    "    -------\n",
    "    list of tuples\n",
    "        A flat list of (ontology type, ontology id).\n",
    "    \"\"\"\n",
    "    result_identifiers = []\n",
    "    for item in items_list:\n",
    "        # Extract identifiers from \"urn:miriam\" URIs (urn:miriam:TYPE:ID)\n",
    "        miriam_matches = re.findall(r'urn:miriam:([^\"]+)\"', item)\n",
    "        for r in miriam_matches:\n",
    "            if ':' in r:\n",
    "                ontology_type, ontology_id = r.split(\":\", 1)\n",
    "                result_identifiers.append((ontology_type, ontology_id))\n",
    "        # Extract identifiers from identifiers.org URLs (identifiers.org/TYPE/ID)\n",
    "        idorg_matches = re.findall(r'identifiers\\.org/([^\"/]+)/([^\"/]+)', item)\n",
    "        for ontology_type, ontology_id in idorg_matches:\n",
    "            result_identifiers.append((ontology_type, ontology_id))\n",
    "    return result_identifiers\n",
    "\n",
    "def divideExistingAnnotation(inp_str, qualifier):\n",
    "    \"\"\"\n",
    "    Parse the annotation string to extract items in multiple <rdf:Bag> elements \n",
    "    under the specified qualifier (e.g., <bqbiol:is>), and keep the rest in the container.\n",
    "    Ensures namespaces and structure are preserved when creating an empty qualifier block.\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    inp_str: str\n",
    "        The full annotation string.\n",
    "    qualifier: str\n",
    "        The qualifier to target (e.g., bqbiol:is, bqbiol:isDescribedBy).\n",
    "\n",
    "    Returns\n",
    "    -------\n",
    "    dict:\n",
    "        - 'container': The annotation string with the qualifier blocks replaced by a single empty block.\n",
    "        - 'items': A list of <rdf:li> elements found in all <rdf:Bag> containers of the specified qualifier.\n",
    "        - If no qualifier blocks are found, returns None.\n",
    "    \"\"\"\n",
    "\n",
    "    # Regex to match all blocks for the specified qualifier, including attributes\n",
    "    qualifier_pattern = rf\"(<{qualifier}\\b[^>]*?>\\s*<rdf:Bag>.*?</rdf:Bag>\\s*</{qualifier}>)\"\n",
    "    qualifier_matches = re.findall(qualifier_pattern, inp_str, re.DOTALL)\n",
    "\n",
    "    if not qualifier_matches:\n",
    "        return None  # Return None if no blocks for the qualifier are found\n",
    "\n",
    "    # Collect all <rdf:li> elements from each matched block\n",
    "    rdf_li_pattern = r\"<rdf:li[^>]*\\/>\"\n",
    "    items = []\n",
    "    for block in qualifier_matches:\n",
    "        items.extend(re.findall(rdf_li_pattern, block))\n",
    "\n",
    "    # Extract the opening tag with attributes from the first match\n",
    "    match_prefix = re.match(rf\"<{qualifier}.*?>\", qualifier_matches[0])\n",
    "    if match_prefix:\n",
    "        qualifier_opening = match_prefix.group()  # Capture opening tag with attributes\n",
    "    else:\n",
    "        qualifier_opening = f\"<{qualifier}>\"\n",
    "\n",
    "    # Construct a single empty qualifier block using the preserved attributes\n",
    "    empty_qualifier_block = (\n",
    "        f\"      {qualifier_opening}\\n\"\n",
    "        f\"        <rdf:Bag>\\n\"\n",
    "        f\"        </rdf:Bag>\\n\"\n",
    "        f\"      </{qualifier}>\"\n",
    "    )\n",
    "    # Remove all original qualifier blocks from the container\n",
    "    stripped_annotation = re.sub(qualifier_pattern, \"\", inp_str, flags=re.DOTALL).strip()\n",
    "\n",
    "    # Reinsert the empty qualifier block with namespaces added\n",
    "    if RDF_TAG not in stripped_annotation:\n",
    "        container = stripped_annotation.replace(\n",
    "            \"<rdf:RDF\",\n",
    "            f\"<{RDF_TAG}\",\n",
    "        )\n",
    "    else:\n",
    "        container = stripped_annotation\n",
    "\n",
    "    container = container.replace(\n",
    "        \"</rdf:Description>\",\n",
    "        f\"{empty_qualifier_block}\\n    </rdf:Description>\"\n",
    "    )\n",
    "    # Remove repeated blank lines\n",
    "    container = re.sub(r\"\\n\\s*\\n\", \"\\n\", container)\n",
    "    return {\"container\": container, \"items\": items}\n",
    "\n",
    "def getOntologyFromString(string_annotation, description=False):\n",
    "    \"\"\"\n",
    "    Parse string and return string annotation,\n",
    "    marked as <bqbiol:is> or <bqbiol:isVersionOf>;\n",
    "    (and extract the description using <bqbiol:isDescribedBy>)\n",
    "    If neither exists, return None.\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    string_annotation: str\n",
    "    description: bool\n",
    "        If True, also extract the description using <bqbiol:isDescribedBy>.\n",
    "\n",
    "    Returns\n",
    "    -------\n",
    "    list-tuple (ontology type, ontology id)\n",
    "         Return [] if none is provided.\n",
    "    \"\"\"\n",
    "\n",
    "    # Define the bqbiol qualifiers to search\n",
    "    bqbiol_qualifiers = ['is', 'isVersionOf']\n",
    "    if description:\n",
    "        bqbiol_qualifiers.append('isDescribedBy')\n",
    "\n",
    "    result_identifiers = []\n",
    "    for one_qualifier in bqbiol_qualifiers:\n",
    "        res = divideExistingAnnotation(string_annotation, 'bqbiol:' + one_qualifier)\n",
    "        if res is None:\n",
    "            continue\n",
    "        items = res['items']\n",
    "        res = extract_ontology_from_items(items)\n",
    "        # remove empty lists and duplicates\n",
    "        res = [r for r in res if r is not None and r != [] and r not in result_identifiers]\n",
    "        result_identifiers.extend(res)\n",
    "\n",
    "    return result_identifiers\n",
    "def collect_ontology_statistics(model_dir, overall_output_csv, model_ontology_csv, description = False):\n",
    "    \"\"\"\n",
    "    Collect ontology statistics of SBML models in a folder.\n",
    "\n",
    "    Parameters:\n",
    "    - folder_path (str): Path to the folder containing SBML models.\n",
    "    - overall_output_csv (str): Path to save overall statistics as a CSV.\n",
    "    - model_ontology_csv (str): Path to save per-model statistics as a CSV.\n",
    "    - description (bool): If True, extract the description using <bqbiol:isDescribedBy>\n",
    "    \"\"\"\n",
    "    overall_counter = Counter()\n",
    "    model_data = []\n",
    "\n",
    "    # Iterate over all SBML files in the folder\n",
    "    model_files = [f for f in os.listdir(model_dir) if f.endswith('.xml') or f.endswith('.sbml')]\n",
    "    for idx, file_name in enumerate(model_files):\n",
    "        print(f\"We are at {idx}: Processing {file_name}\")\n",
    "        file_path = os.path.join(model_dir, file_name)\n",
    "        gene_annotations = get_gene_annotation_string(file_path)\n",
    "\n",
    "        model_counter = Counter()\n",
    "        for gene_annotation in gene_annotations:\n",
    "            resources = getOntologyFromString(gene_annotation, description = description)\n",
    "            # Count ontology types\n",
    "            if resources is not None and resources != []:\n",
    "                for ontology_type, ontology_id in resources:\n",
    "                    overall_counter[ontology_type] += 1\n",
    "                    model_counter[ontology_type] += 1\n",
    "\n",
    "        # Store model-level data\n",
    "        for ontology_type, count in model_counter.items():\n",
    "            model_data.append({\n",
    "                'model': file_name,\n",
    "                'ontology_type': ontology_type,\n",
    "                'count': count\n",
    "            })\n",
    "\n",
    "    # Convert overall statistics to a DataFrame\n",
    "    overall_stats_df = pd.DataFrame.from_dict(overall_counter, orient='index', columns=['count'])\n",
    "    overall_stats_df.reset_index(inplace=True)\n",
    "    overall_stats_df.columns = ['ontology_type', 'count']\n",
    "\n",
    "    # Convert model-specific data to a DataFrame\n",
    "    model_stats_df = pd.DataFrame(model_data)\n",
    "\n",
    "    # Save the dataframes to CSV\n",
    "    overall_stats_df.to_csv(overall_output_csv, index=False)\n",
    "    model_stats_df.to_csv(model_ontology_csv, index=False)\n",
    "    print(f\"Overall statistics saved to {overall_output_csv}\")\n",
    "    print(f\"Model-specific statistics saved to {model_ontology_csv}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We are at 0: Processing 054_PC12-CELL-DIFFERENTIATION_source.sbml\n",
      "We are at 1: Processing 038_SKBR3-BREAST-CELL-LINE-LONG-TERM_source.sbml\n",
      "We are at 2: Processing 073_LYMPHOID-AND-MYELOID-CELL-SPECIFICATION_source.sbml\n",
      "We are at 3: Processing 126_ORF3A_source.sbml\n",
      "We are at 4: Processing 043_BORDETELLA-BRONCHISEPTICA_source.sbml\n",
      "We are at 5: Processing 239_M1-SYNOVIAL-MACROPHAGE_source.sbml\n",
      "We are at 6: Processing 161_DIFFERENTIATION-OF-MONOCYTES_source.sbml\n",
      "We are at 7: Processing 078_IMMUNE-SYSTEM_source.sbml\n",
      "We are at 8: Processing 009_YEAST-APOPTOSIS_source.sbml\n",
      "We are at 9: Processing 072_LYMPHOPOIESIS-REGULATORY-NETWORK_source.sbml\n",
      "We are at 10: Processing 065_TUMOUR-CELL-INVASION-AND-MIGRATION_source.sbml\n",
      "We are at 11: Processing 198_PAIR-RULE-MODULE_source.sbml\n",
      "We are at 12: Processing 090_MAPK-REDUCED-2_source.sbml\n",
      "We are at 13: Processing 179_MICROENVIRONMENT-CONTROL_source.sbml\n",
      "We are at 14: Processing 029_TOLL-PATHWAY-OF-DROSOPHILA_source.sbml\n",
      "We are at 15: Processing 010_CARDIAC-DEVELOPMENT_source.sbml\n",
      "We are at 16: Processing 172_DROSOPHILA-GAP-D_source.sbml\n",
      "We are at 17: Processing 193_SENESCENCE-G1S-CHECKPOINT_source.sbml\n",
      "We are at 18: Processing 123_NSP4-NSP6_source.sbml\n",
      "We are at 19: Processing 053_PREDICTING-VARIABILITIES-IN-CARDIAC-GENE_source.sbml\n",
      "We are at 20: Processing 076_SENESCENCE-ASSOCIATED-SECRETORY-PHENOTYPE_source.sbml\n",
      "We are at 21: Processing 118_INTERFERON-1_source.sbml\n",
      "We are at 22: Processing 020_APOPTOSIS-NETWORK_source.sbml\n",
      "We are at 23: Processing 066_CD4-T-CELL-DIFFERENTIATION_source.sbml\n",
      "We are at 24: Processing 183_ALTERATIONS-IN-BLADDER_source.sbml\n",
      "We are at 25: Processing 178_MAST-CELL-ACTIVATION_source.sbml\n",
      "We are at 26: Processing 190_BRAF-TREATMENT-RESPONSE_source.sbml\n",
      "We are at 27: Processing 218_TCA-CYCLE_source.sbml\n",
      "We are at 28: Processing 047_FGF-PATHWAY-OF-DROSOPHILA_source.sbml\n",
      "We are at 29: Processing 240_M2-SYNOVIAL-MACROPHAGE_source.sbml\n",
      "We are at 30: Processing 169_DROSOPHILA-GAP-A_source.sbml\n",
      "We are at 31: Processing 209_ABERRANT-CELL-CYCLE-PROGRESSION_source.sbml\n",
      "We are at 32: Processing 077_SIGNALLING-PATHWAY-FOR-BUTANOL-PRODUCTION_source.sbml\n",
      "We are at 33: Processing 141_HIGH-OSMOLARITY-AND-MATING-PATHWAYS_source.sbml\n",
      "We are at 34: Processing 145_MELANOGENESIS_source.sbml\n",
      "We are at 35: Processing 238_DNA-DAMAGE-INDUCED-FERROPTOSIS_source.sbml\n",
      "We are at 36: Processing 104_DROSOPHILA-CELL-CYCLE_source.sbml\n",
      "We are at 37: Processing 184_P53-MDM2-NETWORK_source.sbml\n",
      "We are at 38: Processing 018_EGFR-ERBB-SIGNALING_source.sbml\n",
      "We are at 39: Processing 176_MYELOFIBROTIC-MICROENVIRONMENT_source.sbml\n",
      "We are at 40: Processing 151_TCR-REDOX-METABOLISM_source.sbml\n",
      "We are at 41: Processing 132_VIRUS-REPLICATION-CYCLE_source.sbml\n",
      "We are at 42: Processing 046_B-BRONCHISEPTICA-AND-T-RETORTAEFORMIS_source.sbml\n",
      "We are at 43: Processing 016_IL-1-SIGNALING_source.sbml\n",
      "We are at 44: Processing 036_HCC1954-BREAST-CELL-LINE-SHORT-TERM_source.sbml\n",
      "We are at 45: Processing 040_T-CELL-DIFFERENTIATION_source.sbml\n",
      "We are at 46: Processing 283_EMT-MECHANOSENSING-TGFB_source.sbml\n",
      "We are at 47: Processing 191_SEGMENT-POLARITY-1-CELL_source.sbml\n",
      "We are at 48: Processing 122_NSP14_source.sbml\n",
      "We are at 49: Processing 200_LUNG-CANCER-CELL-CYCLE_source.sbml\n",
      "We are at 50: Processing 156_CONTROL-OF-TH1-TH2-TH17-TREG-DIFFERENTATION-REDUCED_source.sbml\n",
      "We are at 51: Processing 034_HCC1954-BREAST-CELL-LINE-LONG-TERM_source.sbml\n",
      "We are at 52: Processing 022_B-CELL-DIFFERENTIATION_source.sbml\n",
      "We are at 53: Processing 035_BT474-BREAST-CELL-LINE-SHORT-TERM_source.sbml\n",
      "We are at 54: Processing 082_TCR-TLR5-SIGNALING-2018_source.sbml\n",
      "We are at 55: Processing 007_CORTICAL-AREA-DEVELOPMENT_source.sbml\n",
      "We are at 56: Processing 189_TRP-BIOSYNTHESIS_source.sbml\n",
      "We are at 57: Processing 274_MERGE-HEMATOPOIESIS-177-271_source.sbml\n",
      "We are at 58: Processing 070_MAPK-CANCER-CELL-FATE_source.sbml\n",
      "We are at 59: Processing 155_CONTROL-OF-TH1-TH2-TH17-TREG-DIFFERENTATION_source.sbml\n",
      "We are at 60: Processing 061_TUMOR-MICROENVIRONMENT-IN-LYMPHOBLASTIC-LEUKEAMIA_source.sbml\n",
      "We are at 61: Processing 241_RHEUMATOID-ARTHRITIS-TH1_source.sbml\n",
      "We are at 62: Processing 139_ACUTE-RESPONSES-DURING-HYPERINSULINEMIA_source.sbml\n",
      "We are at 63: Processing 069_IRON-ACQUISITION-AND-STRESS-RESPONSE_source.sbml\n",
      "We are at 64: Processing 106_STOMATAL-RESTING-STATE_source.sbml\n",
      "We are at 65: Processing 159_BUDDING-YEAST-CORE_source.sbml\n",
      "We are at 66: Processing 095_FISSION-YEAST-2008_source.sbml\n",
      "We are at 67: Processing 175_SEA-URCHIN_source.sbml\n",
      "We are at 68: Processing 063_LAC-OPERON_source.sbml\n",
      "We are at 69: Processing 013_CHOLESTEROL-REGULATORY-PATHWAY_source.sbml\n",
      "We are at 70: Processing 032_T-CELL-SIGNALLING-2006_source.sbml\n",
      "We are at 71: Processing 097_DROSOPHILA-WINGS-AP_source.sbml\n",
      "We are at 72: Processing 214_FOXO3-PATHWAY_source.sbml\n",
      "We are at 73: Processing 165_EGGSHELL-PATTERNING-PHENOMOENOLOGICAL_source.sbml\n",
      "We are at 74: Processing 187_MAMMAL-SEX-DETERMINATION-1-CELL_source.sbml\n",
      "We are at 75: Processing 136_EGF-TNF-ALPHA-SIGNALLING-PATHWAY_source.sbml\n",
      "We are at 76: Processing 131_TGFB-PATHWAY_source.sbml\n",
      "We are at 77: Processing 015_NEUROTRANSMITTER-SIGNALING-PATHWAY_source.sbml\n",
      "We are at 78: Processing 216_PPARGC1A-PATHWAY_source.sbml\n",
      "We are at 79: Processing 219_WNT-PI3K-AKT-PATHWAY_source.sbml\n",
      "We are at 80: Processing 135_SIGNAL-TRANSDUCTION_source.sbml\n",
      "We are at 81: Processing 203_DDR-SIGNALLING-PATHWAYS_source.sbml\n",
      "We are at 82: Processing 088_MIR-9-NEUROGENESIS_source.sbml\n",
      "We are at 83: Processing 277_HYPOXIA-AND-MECHANOSENSING-IN-METASTASIS_source.sbml\n",
      "We are at 84: Processing 108_GEROCONVERSION_source.sbml\n",
      "We are at 85: Processing 196_T-LYMPHOCYTE-SPECIFICATION_source.sbml\n",
      "We are at 86: Processing 003_MAMMALIAN-CELL-CYCLE_source.sbml\n",
      "We are at 87: Processing 062_CD4-T-CELL-DIFFERENTIATION-AND-PLASTICITY_source.sbml\n",
      "We are at 88: Processing 042_TOL-REGULATORY-NETWORK_source.sbml\n",
      "We are at 89: Processing 099_YEAST-HYPHAL-TRANSITION_source.sbml\n",
      "We are at 90: Processing 120_KYNURENINE-PATHWAY_source.sbml\n",
      "We are at 91: Processing 071_CASTRATION-RESISTANT-PROSTATE-CANCER_source.sbml\n",
      "We are at 92: Processing 005_FA-BRCA-PATHWAY_source.sbml\n",
      "We are at 93: Processing 021_BODY-SEGMENTATION-IN-DROSOPHILA-2013_source.sbml\n",
      "We are at 94: Processing 164_EGGSHELL-PATTERNING-MECHANISTIC_source.sbml\n",
      "We are at 95: Processing 195_CTLA4-PD1-CHECKPOINT-INHIBITORS_source.sbml\n",
      "We are at 96: Processing 019_IL-6-SIGNALING_source.sbml\n",
      "We are at 97: Processing 166_DROSOPHILA-JAK-STAT-PATHWAY_source.sbml\n",
      "We are at 98: Processing 059_BORTEZOMIB-RESPONSES-IN-MYELOMA-CELLS_source.sbml\n",
      "We are at 99: Processing 181_MULTILEVEL-CELL-CYCLE_source.sbml\n",
      "We are at 100: Processing 008_DEATH-RECEPTOR-SIGNALING_source.sbml\n",
      "We are at 101: Processing 272_FLT3-ITD-AML-SIGNALING_source.sbml\n",
      "We are at 102: Processing 130_RENIN-ANGIOTENSIN_source.sbml\n",
      "We are at 103: Processing 148_AGS-CELL-FATE-DECISION_source.sbml\n",
      "We are at 104: Processing 109_ASYMMETRIC-CELL-DIVISION-A_source.sbml\n",
      "We are at 105: Processing 014_T-LGL-SURVIVAL-NETWORK-2008_source.sbml\n",
      "We are at 106: Processing 068_AURORA-KINASE-A-IN-NEUROBLASTOMA_source.sbml\n",
      "We are at 107: Processing 170_DROSOPHILA-GAP-B_source.sbml\n",
      "We are at 108: Processing 282_EMT-MECHANOSENSING_source.sbml\n",
      "We are at 109: Processing 134_RHEUMATOID-ARTHRITIS_source.sbml\n",
      "We are at 110: Processing 152_TCR-REDOX-METABOLISM-REDUCED_source.sbml\n",
      "We are at 111: Processing 186_CHICKEN-SEX-DETERMINATION-REDUCED_source.sbml\n",
      "We are at 112: Processing 004_ERBB-RECEPTOR-SIGNALING_source.sbml\n",
      "We are at 113: Processing 086_TUMOUR-CELL-INVASION-AND-MIGRATION-REDUCED_source.sbml\n",
      "We are at 114: Processing 157_CONTROL-OF-TH-DIFFERENTATION_source.sbml\n",
      "We are at 115: Processing 188_MAMMAL-SEX-DETERMINATION-2-CELL_source.sbml\n",
      "We are at 116: Processing 064_METABOLIC-INTERACTIONS-IN-GUT-MICROBIOME_source.sbml\n",
      "We are at 117: Processing 057_FANCONI-ANEMIA-AND-CHECKPOINT-RECOVERY_source.sbml\n",
      "We are at 118: Processing 044_TRICHOSTRONGYLUS-RETORTAEFORMIS_source.sbml\n",
      "We are at 119: Processing 031_CELL-CYCLE-TRANSCRIPTION_source.sbml\n",
      "We are at 120: Processing 167_MESODERM-SPECIFICATION-IN-DROSOPHILA_source.sbml\n",
      "We are at 121: Processing 194_VULVAR-PRECURSOR-CELLS_source.sbml\n",
      "We are at 122: Processing 185_CHICKEN-SEX-DETERMINATION_source.sbml\n",
      "We are at 123: Processing 163_DROSOPHILA-EGF-PATHWAY_source.sbml\n",
      "We are at 124: Processing 142_BLOOD-STEM-CELL_source.sbml\n",
      "We are at 125: Processing 149_AGS-CELL-FATE-DECISION-REDUCED_source.sbml\n",
      "We are at 126: Processing 058_ARABIDOPSIS-THALIANA-CELL-CYCLE_source.sbml\n",
      "We are at 127: Processing 143_BREAST-CANCER-INHIBITORS_source.sbml\n",
      "We are at 128: Processing 146_BUDDING-YEAST-FAURE-2009_source.sbml\n",
      "We are at 129: Processing 133_ROOT-STEM-CELL-2010_source.sbml\n",
      "We are at 130: Processing 055_HUMAN-GONADAL-SEX-DETERMINATION_source.sbml\n",
      "We are at 131: Processing 213_DOPAMINE-TRANSCRIPTION_source.sbml\n",
      "We are at 132: Processing 085_REPROGRAMMING-TESTES-DERIVED-STEM-CELLS_source.sbml\n",
      "We are at 133: Processing 096_ERBB-REGULATED-G1-S-TRANSITION_source.sbml\n",
      "We are at 134: Processing 111_APOPTOSIS_source.sbml\n",
      "We are at 135: Processing 017_DIFFERENTIATION-OF-T-LYMPHOCYTES_source.sbml\n",
      "We are at 136: Processing 037_SKBR3-BREAST-CELL-LINE-SHORT-TERM_source.sbml\n",
      "We are at 137: Processing 110_ASYMMETRIC-CELL-DIVISION-B_source.sbml\n",
      "We are at 138: Processing 039_HIV-1-INTERACTIONS-WITH-T-CELL-SIGNALING_source.sbml\n",
      "We are at 139: Processing 116_HMOX1-PATHWAY_source.sbml\n",
      "We are at 140: Processing 006_HGF-SIGNALING-IN-KERATINOCYTES_source.sbml\n",
      "We are at 141: Processing 162_DROSOPHILA-DPP-PATHWAY_source.sbml\n",
      "We are at 142: Processing 045_HH-PATHWAY-OF-DROSOPHILA_source.sbml\n",
      "We are at 143: Processing 284_ANCHORAGE-DEPENDENCE-PROLIFERATION_source.sbml\n",
      "We are at 144: Processing 081_TLR5-SIGNALING-2018_source.sbml\n",
      "We are at 145: Processing 049_OXIDATIVE-STRESS-PATHWAY_source.sbml\n",
      "We are at 146: Processing 147_BUDDING-YEAST-EXIT-MODULE_source.sbml\n",
      "We are at 147: Processing 150_CELL-FATE-DECISION-MULTISCALE_source.sbml\n",
      "We are at 148: Processing 125_ORF10-CUL2-PATHWAY_source.sbml\n",
      "We are at 149: Processing 128_PYRIMIDINE-DEPRIVATION_source.sbml\n",
      "We are at 150: Processing 113_ER-STRESS_source.sbml\n",
      "We are at 151: Processing 280_SHH-SIGNALING-EMT-REGULATION_source.sbml\n",
      "We are at 152: Processing 023_MAMMALIAN-CELL-CYCLE-2006_source.sbml\n",
      "We are at 153: Processing 115_E-PROTEIN_source.sbml\n",
      "We are at 154: Processing 025_T-LGL-SURVIVAL-NETWORK-2011_source.sbml\n",
      "We are at 155: Processing 137_SIGNALLING-IN-LIVER-CANCER_source.sbml\n",
      "We are at 156: Processing 093_IMMUNE-CHECKPOINT-INHIBITORS_source.sbml\n",
      "We are at 157: Processing 041_INFLUENZA-VIRUS-REPLICATION-CYCLE_source.sbml\n",
      "We are at 158: Processing 119_JNK-PATHWAY_source.sbml\n",
      "We are at 159: Processing 067_REGULATION-OF-L-ARABINOSE-OPERON_source.sbml\n",
      "We are at 160: Processing 083_SIGNALING-IN-PROSTATE-CANCER_source.sbml\n",
      "We are at 161: Processing 153_CONTROL-OF-PROLIFERATION_source.sbml\n",
      "We are at 162: Processing 182_BOOLEAN-CELL-CYCLE_source.sbml\n",
      "We are at 163: Processing 117_IFN-LAMBDA_source.sbml\n",
      "We are at 164: Processing 089_MAPK-REDUCED-1_source.sbml\n",
      "We are at 165: Processing 091_MAPK-REDUCED-3_source.sbml\n",
      "We are at 166: Processing 242_RHEUMATOID-ARTHRITIS-FIBROBLAST_source.sbml\n",
      "We are at 167: Processing 060_STOMATAL-OPENING_source.sbml\n",
      "We are at 168: Processing 028_VEGF-PATHWAY-OF-DROSOPHILA_source.sbml\n",
      "We are at 169: Processing 048_GLUCOSE-REPRESSION-SIGNALING-2009_source.sbml\n",
      "We are at 170: Processing 197_ANTERIOR-POSTERIOR-BOUNDARY_source.sbml\n",
      "We are at 171: Processing 080_TCR-SIGNALING-2018_source.sbml\n",
      "We are at 172: Processing 171_DROSOPHILA-GAP-C_source.sbml\n",
      "We are at 173: Processing 050_CD4-T-CELL-SIGNALING_source.sbml\n",
      "We are at 174: Processing 158_LAMBDA-PHAGE-LYSOGENY_source.sbml\n",
      "We are at 175: Processing 204_HUMAN-BRAIN-ORGANOIDS_source.sbml\n",
      "We are at 176: Processing 112_COAGULATION-PATHWAY_source.sbml\n",
      "We are at 177: Processing 124_NSP9-PROTEIN_source.sbml\n",
      "We are at 178: Processing 154_CONTROL-OF-TH1-TH2-DIFFERENTIATION_source.sbml\n",
      "We are at 179: Processing 168_DROSOPHILA-NOTCH-PATHWAY_source.sbml\n",
      "We are at 180: Processing 012_T-CELL-RECEPTOR-SIGNALING_source.sbml\n",
      "We are at 181: Processing 052_SEPTATION-INITIATION-NETWORK_source.sbml\n",
      "We are at 182: Processing 215_MTOR-PATHWAY_source.sbml\n",
      "We are at 183: Processing 026_BUDDING-YEAST-CELL-CYCLE-2009_source.sbml\n",
      "We are at 184: Processing 217_PRKN-PATHWAY_source.sbml\n",
      "We are at 185: Processing 114_ETC_source.sbml\n",
      "We are at 186: Processing 002_SIGNAL-TRANSDUCTION-IN-FIBROBLASTS_source.sbml\n",
      "We are at 187: Processing 127_PAMP-SIGNALING_source.sbml\n",
      "We are at 188: Processing 180_MORPHOGENETIC-CHECKPOINT_source.sbml\n",
      "We are at 189: Processing 051_COLITIS-ASSOCIATED-COLON-CANCER_source.sbml\n",
      "We are at 190: Processing 160_IL17-DIFFERENTIAL-EXPRESSION_source.sbml\n",
      "We are at 191: Processing 001_SIGNALING-IN-MACROPHAGE-ACTIVATION_source.sbml\n",
      "We are at 192: Processing 056_IGVH-MUTATIONS-IN-LYMPHOCYTIC-LEUKEMIA_source.sbml\n",
      "We are at 193: Processing 243_RHEUMATOID-ARTHRITIS-MULTI-CELLULAR_source.sbml\n",
      "We are at 194: Processing 033_BT474-BREAST-CELL-LINE-LONG-TERM_source.sbml\n",
      "We are at 195: Processing 129_RTC-AND-TRANSCRIPTION_source.sbml\n",
      "We are at 196: Processing 192_SEGMENT-POLARITY-6-CELL_source.sbml\n",
      "We are at 197: Processing 030_SPZ-NETWORK-OF-DROSOPHILA_source.sbml\n",
      "We are at 198: Processing 024_BUDDING-YEAST-CELL-CYCLE_source.sbml\n",
      "We are at 199: Processing 285_MITOCHONDRIAL-DISFUNCTION_source.sbml\n",
      "We are at 200: Processing 074_T-LGL-SURVIVAL-NETWORK-2011-REDUCED_source.sbml\n",
      "We are at 201: Processing 027_WG-PATHWAY-OF-DROSOPHILA_source.sbml\n",
      "We are at 202: Processing 011_GUARD-CELL-ABSCISIC-ACID-SIGNALING_source.sbml\n",
      "We are at 203: Processing 275_MERGE-AML-PROGRESSION-100-273_source.sbml\n",
      "We are at 204: Processing 121_NLRP3-ACTIVATION_source.sbml\n",
      "Overall statistics saved to /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/biodivine_260125_ontology_statistics.csv\n",
      "Model-specific statistics saved to /Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/biodivine_260125_model_ontology_statistics.csv\n"
     ]
    }
   ],
   "source": [
    "output_csv = biodivine_model_dir+\"biodivine_260125_ontology_statistics.csv\"\n",
    "model_ontology_csv = biodivine_model_dir+\"biodivine_260125_model_ontology_statistics.csv\"\n",
    "collect_ontology_statistics(biodivine_model_dir, output_csv, model_ontology_csv, description = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of models with annotation: 86\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>model</th>\n",
       "      <th>ontology_type</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>038_SKBR3-BREAST-CELL-LINE-LONG-TERM_source.sbml</td>\n",
       "      <td>pubmed</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>126_ORF3A_source.sbml</td>\n",
       "      <td>pubmed</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>043_BORDETELLA-BRONCHISEPTICA_source.sbml</td>\n",
       "      <td>pubmed</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>239_M1-SYNOVIAL-MACROPHAGE_source.sbml</td>\n",
       "      <td>pubmed</td>\n",
       "      <td>588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>078_IMMUNE-SYSTEM_source.sbml</td>\n",
       "      <td>pubmed</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>288</th>\n",
       "      <td>033_BT474-BREAST-CELL-LINE-LONG-TERM_source.sbml</td>\n",
       "      <td>pubmed</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>289</th>\n",
       "      <td>030_SPZ-NETWORK-OF-DROSOPHILA_source.sbml</td>\n",
       "      <td>pubmed</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290</th>\n",
       "      <td>024_BUDDING-YEAST-CELL-CYCLE_source.sbml</td>\n",
       "      <td>pubmed</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>291</th>\n",
       "      <td>027_WG-PATHWAY-OF-DROSOPHILA_source.sbml</td>\n",
       "      <td>pubmed</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>292</th>\n",
       "      <td>011_GUARD-CELL-ABSCISIC-ACID-SIGNALING_source....</td>\n",
       "      <td>pubmed</td>\n",
       "      <td>121</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>76 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 model ontology_type  count\n",
       "0     038_SKBR3-BREAST-CELL-LINE-LONG-TERM_source.sbml        pubmed     71\n",
       "3                                126_ORF3A_source.sbml        pubmed      2\n",
       "5            043_BORDETELLA-BRONCHISEPTICA_source.sbml        pubmed     42\n",
       "6               239_M1-SYNOVIAL-MACROPHAGE_source.sbml        pubmed    588\n",
       "18                       078_IMMUNE-SYSTEM_source.sbml        pubmed     31\n",
       "..                                                 ...           ...    ...\n",
       "288   033_BT474-BREAST-CELL-LINE-LONG-TERM_source.sbml        pubmed     72\n",
       "289          030_SPZ-NETWORK-OF-DROSOPHILA_source.sbml        pubmed     46\n",
       "290           024_BUDDING-YEAST-CELL-CYCLE_source.sbml        pubmed     57\n",
       "291           027_WG-PATHWAY-OF-DROSOPHILA_source.sbml        pubmed     61\n",
       "292  011_GUARD-CELL-ABSCISIC-ACID-SIGNALING_source....        pubmed    121\n",
       "\n",
       "[76 rows x 3 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model_ontology = pd.read_csv(biodivine_model_dir+\"biodivine_260125_model_ontology_statistics.csv\")\n",
    "print(f\"Total number of models with annotation: {model_ontology['model'].nunique()}\")\n",
    "model_ontology[model_ontology['ontology_type'] == 'pubmed']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<annotation>\n",
      "  <rdf:RDF>\n",
      "    <rdf:Description rdf:about=\"#s_id_sa83\">\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc.symbol:BCL2\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc.symbol:BCL2\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:refseq:NM_000657\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ncbigene:596\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ncbigene:596\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:uniprot:P10415\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:uniprot:P10415\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ensembl:ENSG00000171791\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:refseq:NM_000633\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc:990\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <dcterms:modified rdf:parseType=\"Resource\"/>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ncbigene:4170\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ncbigene:4170\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:uniprot:Q07820\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:uniprot:Q07820\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc:6943\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:refseq:NM_021960\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ensembl:ENSG00000143384\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc.symbol:MCL1\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc.symbol:MCL1\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <dcterms:modified rdf:parseType=\"Resource\"/>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:refseq:NM_138578\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ncbigene:598\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ncbigene:598\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ensembl:ENSG00000171552\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:uniprot:Q07817\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:uniprot:Q07817\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc:992\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc.symbol:BCL2L1\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc.symbol:BCL2L1\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <dcterms:modified rdf:parseType=\"Resource\"/>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc.symbol:BCL2\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc.symbol:BCL2\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:refseq:NM_000657\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ncbigene:596\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ncbigene:596\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:uniprot:P10415\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:uniprot:P10415\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ensembl:ENSG00000171791\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:refseq:NM_000633\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc:990\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <dcterms:modified rdf:parseType=\"Resource\"/>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:refseq:NM_138578\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ncbigene:598\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ncbigene:598\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ensembl:ENSG00000171552\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:uniprot:Q07817\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:uniprot:Q07817\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc:992\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc.symbol:BCL2L1\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc.symbol:BCL2L1\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <dcterms:modified rdf:parseType=\"Resource\"/>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ncbigene:4170\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ncbigene:4170\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:uniprot:Q07820\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:uniprot:Q07820\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc:6943\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:refseq:NM_021960\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:ensembl:ENSG00000143384\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc.symbol:MCL1\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:miriam:hgnc.symbol:MCL1\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <dcterms:modified rdf:parseType=\"Resource\"/>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:casq:function:!Orf7a&amp;!E&amp;!BAD\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "      <bqbiol:isDescribedBy>\n",
      "        <rdf:Bag>\n",
      "          <rdf:li rdf:resource=\"urn:casq:cdid:s_id_csa11\"/>\n",
      "        </rdf:Bag>\n",
      "      </bqbiol:isDescribedBy>\n",
      "    </rdf:Description>\n",
      "  </rdf:RDF>\n",
      "</annotation>\n"
     ]
    }
   ],
   "source": [
    "sbml = \"/Users/luna/Desktop/CRBM/AMAS_proj/Models/BioDivine_260125/111_APOPTOSIS_source.sbml\"\n",
    "reader = libsbml.SBMLReader()\n",
    "document = reader.readSBML(sbml)\n",
    "model = document.getModel()\n",
    "qual_spec = model.getPlugin(\"qual\").getQualitativeSpecies('csa11')\n",
    "print(qual_spec.getAnnotationString())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run Converter\n",
    "\n",
    "Test TabularQual converter by converting SBML â†’ Spreadsheet â†’ SBML and comparing counts.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import libsbml\n",
    "import pandas as pd\n",
    "import re\n",
    "from collections import Counter\n",
    "import warnings\n",
    "\n",
    "# Import TabularQual converter\n",
    "sys.path.insert(0, '/Users/luna/Desktop/CRBM/Standards/TabularQual')\n",
    "from TabularQual_converter.convert_sbml_to_spreadsheet import convert_sbml_to_spreadsheet\n",
    "from TabularQual_converter.convert_spreadsheet_to_sbml import convert_spreadsheet_to_sbml\n",
    "from TabularQual_converter.sbml_reader import read_sbml\n",
    "\n",
    "# Create output directories\n",
    "spreadsheet_output_dir = os.path.join(biodivine_model_dir, \"spreadsheets\")\n",
    "sbml_output_dir = os.path.join(biodivine_model_dir, \"converted_sbml\")\n",
    "os.makedirs(spreadsheet_output_dir, exist_ok=True)\n",
    "os.makedirs(sbml_output_dir, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_sbml_stats(sbml_path):\n",
    "    \"\"\"\n",
    "    Count species, transitions, and annotations from an SBML-qual file.\n",
    "    \n",
    "    Returns:\n",
    "        dict with keys: species_count, transitions_count, \n",
    "            species_annotations_count, transition_annotations_count\n",
    "    \"\"\"\n",
    "    try:\n",
    "        reader = libsbml.SBMLReader()\n",
    "        document = reader.readSBML(sbml_path)\n",
    "        model = document.getModel()\n",
    "        if model is None:\n",
    "            return {\"error\": \"No model found\"}\n",
    "        \n",
    "        qual_plugin = model.getPlugin(\"qual\")\n",
    "        if not qual_plugin:\n",
    "            return {\"error\": \"No qual plugin found\"}\n",
    "        \n",
    "        # Count species and their annotations\n",
    "        species_count = qual_plugin.getNumQualitativeSpecies()\n",
    "        species_annotations = 0\n",
    "        for i in range(species_count):\n",
    "            qs = qual_plugin.getQualitativeSpecies(i)\n",
    "            if qs.isSetAnnotation():\n",
    "                anno_str = qs.getAnnotationString()\n",
    "                # Count number of rdf:li elements (each is an annotation)\n",
    "                species_annotations += len(re.findall(r'<rdf:li[^>]*resource=', anno_str))\n",
    "        \n",
    "        # Count transitions and their annotations\n",
    "        transitions_count = qual_plugin.getNumTransitions()\n",
    "        transition_annotations = 0\n",
    "        for i in range(transitions_count):\n",
    "            tr = qual_plugin.getTransition(i)\n",
    "            if tr.isSetAnnotation():\n",
    "                anno_str = tr.getAnnotationString()\n",
    "                transition_annotations += len(re.findall(r'<rdf:li[^>]*resource=', anno_str))\n",
    "        \n",
    "        return {\n",
    "            \"species_count\": species_count,\n",
    "            \"transitions_count\": transitions_count,\n",
    "            \"species_annotations\": species_annotations,\n",
    "            \"transition_annotations\": transition_annotations,\n",
    "            \"error\": None\n",
    "        }\n",
    "    except Exception as e:\n",
    "        return {\"error\": str(e)}\n",
    "\n",
    "\n",
    "def count_spreadsheet_stats(xlsx_path):\n",
    "    \"\"\"\n",
    "    Count species, transitions, and annotations from a spreadsheet.\n",
    "    Uses TabularQual's read_spreadsheet_to_model function.\n",
    "    \n",
    "    Returns:\n",
    "        dict with keys: species_count, transitions_count, \n",
    "            species_annotations_count, transition_annotations_count\n",
    "    \"\"\"\n",
    "    try:\n",
    "        from TabularQual_converter.spreadsheet_reader import read_spreadsheet_to_model\n",
    "        \n",
    "        model, _ = read_spreadsheet_to_model(xlsx_path)\n",
    "        \n",
    "        # Count species and their annotations\n",
    "        species_count = len(model.species)\n",
    "        species_annotations = sum(len(sp.annotations) for sp in model.species.values())\n",
    "        \n",
    "        # Count transitions and their annotations\n",
    "        transitions_count = len(model.transitions)\n",
    "        transition_annotations = sum(len(tr.annotations) for tr in model.transitions)\n",
    "        \n",
    "        return {\n",
    "            \"species_count\": species_count,\n",
    "            \"transitions_count\": transitions_count,\n",
    "            \"species_annotations\": species_annotations,\n",
    "            \"transition_annotations\": transition_annotations,\n",
    "            \"error\": None\n",
    "        }\n",
    "    except Exception as e:\n",
    "        return {\"error\": str(e)}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 205 SBML models to process\n",
      "\n",
      "[1/205] Processing: 054_PC12-CELL-DIFFERENTIATION_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "\n",
      "\n",
      "[2/205] Processing: 038_SKBR3-BREAST-CELL-LINE-LONG-TERM_source.sbml\n",
      "Found 25 species\n",
      "Found 21 transitions\n",
      "Found 81 interactions\n",
      "  Converted to spreadsheet in 0.063s\n",
      "Found 25 species\n",
      "Found 21 transitions\n",
      "Found 81 interactions\n",
      "  Converted back to SBML in 0.099s\n",
      "\n",
      "[3/205] Processing: 073_LYMPHOID-AND-MYELOID-CELL-SPECIFICATION_source.sbml\n",
      "Warning: Transition for target 'Cebpa_ER' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 34 species\n",
      "Found 32 transitions\n",
      "Found 96 interactions\n",
      "  Converted to spreadsheet in 0.066s\n",
      "Found 34 species\n",
      "Found 32 transitions\n",
      "Found 96 interactions\n",
      "  Converted back to SBML in 0.109s\n",
      "\n",
      "[4/205] Processing: 126_ORF3A_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[5/205] Processing: 043_BORDETELLA-BRONCHISEPTICA_source.sbml\n",
      "Found 33 species\n",
      "Found 33 transitions\n",
      "Found 79 interactions\n",
      "  Converted to spreadsheet in 0.076s\n",
      "Found 33 species\n",
      "Found 33 transitions\n",
      "Found 79 interactions\n",
      "  Converted back to SBML in 0.102s\n",
      "\n",
      "[6/205] Processing: 239_M1-SYNOVIAL-MACROPHAGE_source.sbml\n",
      "Found 303 species\n",
      "Found 234 transitions\n",
      "Found 562 interactions\n",
      "  Converted to spreadsheet in 0.172s\n",
      "Found 303 species\n",
      "Found 234 transitions\n",
      "Found 562 interactions\n",
      "  Converted back to SBML in 0.190s\n",
      "\n",
      "[7/205] Processing: 161_DIFFERENTIATION-OF-MONOCYTES_source.sbml\n",
      "Found 96 species\n",
      "Found 94 transitions\n",
      "Found 244 interactions\n",
      "  Converted to spreadsheet in 0.081s\n",
      "Found 96 species\n",
      "Found 94 transitions\n",
      "Found 244 interactions\n",
      "  Converted back to SBML in 0.124s\n",
      "\n",
      "[8/205] Processing: 078_IMMUNE-SYSTEM_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[9/205] Processing: 009_YEAST-APOPTOSIS_source.sbml\n",
      "Found 73 species\n",
      "Found 60 transitions\n",
      "Found 114 interactions\n",
      "  Converted to spreadsheet in 0.083s\n",
      "Found 73 species\n",
      "Found 60 transitions\n",
      "Found 114 interactions\n",
      "  Converted back to SBML in 0.106s\n",
      "\n",
      "[10/205] Processing: 072_LYMPHOPOIESIS-REGULATORY-NETWORK_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "\n",
      "\n",
      "[11/205] Processing: 065_TUMOUR-CELL-INVASION-AND-MIGRATION_source.sbml\n",
      "Found 32 species\n",
      "Found 30 transitions\n",
      "Found 156 interactions\n",
      "  Converted to spreadsheet in 0.074s\n",
      "Found 32 species\n",
      "Found 30 transitions\n",
      "Found 156 interactions\n",
      "  Converted back to SBML in 0.096s\n",
      "\n",
      "[12/205] Processing: 198_PAIR-RULE-MODULE_source.sbml\n",
      "Found 11 species\n",
      "Found 11 transitions\n",
      "Found 48 interactions\n",
      "  Converted to spreadsheet in 0.046s\n",
      "Found 11 species\n",
      "Found 11 transitions\n",
      "Found 48 interactions\n",
      "  Converted back to SBML in 0.081s\n",
      "\n",
      "[13/205] Processing: 090_MAPK-REDUCED-2_source.sbml\n",
      "Found 18 species\n",
      "Found 14 transitions\n",
      "Found 60 interactions\n",
      "  Converted to spreadsheet in 0.055s\n",
      "Found 18 species\n",
      "Found 14 transitions\n",
      "Found 60 interactions\n",
      "  Converted back to SBML in 0.091s\n",
      "\n",
      "[14/205] Processing: 179_MICROENVIRONMENT-CONTROL_source.sbml\n",
      "Found 56 species\n",
      "Found 46 transitions\n",
      "Found 149 interactions\n",
      "  Converted to spreadsheet in 0.080s\n",
      "Found 56 species\n",
      "Found 46 transitions\n",
      "Found 149 interactions\n",
      "  Converted back to SBML in 0.114s\n",
      "\n",
      "[15/205] Processing: 029_TOLL-PATHWAY-OF-DROSOPHILA_source.sbml\n",
      "Found 11 species\n",
      "Found 9 transitions\n",
      "Found 11 interactions\n",
      "  Converted to spreadsheet in 0.053s\n",
      "Found 11 species\n",
      "Found 9 transitions\n",
      "Found 11 interactions\n",
      "  Converted back to SBML in 0.107s\n",
      "\n",
      "[16/205] Processing: 010_CARDIAC-DEVELOPMENT_source.sbml\n",
      "Found 15 species\n",
      "Found 13 transitions\n",
      "Found 37 interactions\n",
      "  Converted to spreadsheet in 0.058s\n",
      "Found 15 species\n",
      "Found 13 transitions\n",
      "Found 37 interactions\n",
      "  Converted back to SBML in 0.089s\n",
      "\n",
      "[17/205] Processing: 172_DROSOPHILA-GAP-D_source.sbml\n",
      "Warning: Transition for target 'Hb_b3' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Kr_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 7 species\n",
      "Found 7 transitions\n",
      "Found 12 interactions\n",
      "  Converted to spreadsheet in 0.052s\n",
      "Found 7 species\n",
      "Found 7 transitions\n",
      "Found 12 interactions\n",
      "  Converted back to SBML in 0.085s\n",
      "\n",
      "[18/205] Processing: 193_SENESCENCE-G1S-CHECKPOINT_source.sbml\n",
      "Found 30 species\n",
      "Found 28 transitions\n",
      "Found 100 interactions\n",
      "  Converted to spreadsheet in 0.056s\n",
      "Found 30 species\n",
      "Found 28 transitions\n",
      "Found 100 interactions\n",
      "  Converted back to SBML in 0.103s\n",
      "\n",
      "[19/205] Processing: 123_NSP4-NSP6_source.sbml\n",
      "Found 60 species\n",
      "Found 43 transitions\n",
      "Found 62 interactions\n",
      "  Converted to spreadsheet in 0.079s\n",
      "Found 60 species\n",
      "Found 43 transitions\n",
      "Found 62 interactions\n",
      "  Converted back to SBML in 0.107s\n",
      "\n",
      "[20/205] Processing: 053_PREDICTING-VARIABILITIES-IN-CARDIAC-GENE_source.sbml\n",
      "Found 15 species\n",
      "Found 13 transitions\n",
      "Found 37 interactions\n",
      "  Converted to spreadsheet in 0.053s\n",
      "Found 15 species\n",
      "Found 13 transitions\n",
      "Found 37 interactions\n",
      "  Converted back to SBML in 0.091s\n",
      "\n",
      "[21/205] Processing: 076_SENESCENCE-ASSOCIATED-SECRETORY-PHENOTYPE_source.sbml\n",
      "Found 51 species\n",
      "Found 49 transitions\n",
      "Found 96 interactions\n",
      "  Converted to spreadsheet in 0.076s\n",
      "Found 51 species\n",
      "Found 49 transitions\n",
      "Found 96 interactions\n",
      "  Converted back to SBML in 0.103s\n",
      "\n",
      "[22/205] Processing: 118_INTERFERON-1_source.sbml\n",
      "Found 121 species\n",
      "Found 66 transitions\n",
      "Found 190 interactions\n",
      "  Converted to spreadsheet in 0.103s\n",
      "Found 121 species\n",
      "Found 66 transitions\n",
      "Found 190 interactions\n",
      "  Converted back to SBML in 0.126s\n",
      "\n",
      "[23/205] Processing: 020_APOPTOSIS-NETWORK_source.sbml\n",
      "Found 41 species\n",
      "Found 39 transitions\n",
      "Found 73 interactions\n",
      "  Converted to spreadsheet in 0.082s\n",
      "Found 41 species\n",
      "Found 39 transitions\n",
      "Found 73 interactions\n",
      "  Converted back to SBML in 0.105s\n",
      "\n",
      "[24/205] Processing: 066_CD4-T-CELL-DIFFERENTIATION_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[25/205] Processing: 183_ALTERATIONS-IN-BLADDER_source.sbml\n",
      "Found 35 species\n",
      "Found 31 transitions\n",
      "Found 111 interactions\n",
      "  Converted to spreadsheet in 0.073s\n",
      "Found 35 species\n",
      "Found 31 transitions\n",
      "Found 111 interactions\n",
      "  Converted back to SBML in 0.097s\n",
      "\n",
      "[26/205] Processing: 178_MAST-CELL-ACTIVATION_source.sbml\n",
      "Found 48 species\n",
      "Found 45 transitions\n",
      "Found 72 interactions\n",
      "  Converted to spreadsheet in 0.055s\n",
      "Found 48 species\n",
      "Found 45 transitions\n",
      "Found 72 interactions\n",
      "  Converted back to SBML in 0.086s\n",
      "\n",
      "[27/205] Processing: 190_BRAF-TREATMENT-RESPONSE_source.sbml\n",
      "Found 37 species\n",
      "Found 32 transitions\n",
      "Found 74 interactions\n",
      "  Converted to spreadsheet in 0.052s\n",
      "Found 37 species\n",
      "Found 32 transitions\n",
      "Found 74 interactions\n",
      "  Converted back to SBML in 0.084s\n",
      "\n",
      "[28/205] Processing: 218_TCA-CYCLE_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: The value of the attribute 'qual:sign' of an <input> object must conform to the syntax of the SBML data type 'sign' and may only take on the allowed values of 'sign' defined in SBML; that is, the value must be one of the following: 'positive', 'negative', 'dual' or 'unknown'.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      " The inputSign on the <input> with id 'tr_sa11967_in_0' is 'mixed', which is not a valid option.\n",
      "\n",
      "\n",
      "[29/205] Processing: 047_FGF-PATHWAY-OF-DROSOPHILA_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[30/205] Processing: 240_M2-SYNOVIAL-MACROPHAGE_source.sbml\n",
      "Found 249 species\n",
      "Found 197 transitions\n",
      "Found 430 interactions\n",
      "  Converted to spreadsheet in 0.155s\n",
      "Found 249 species\n",
      "Found 197 transitions\n",
      "Found 430 interactions\n",
      "  Converted back to SBML in 0.159s\n",
      "\n",
      "[31/205] Processing: 169_DROSOPHILA-GAP-A_source.sbml\n",
      "Warning: Transition for target 'Hb_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Kni' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 7 species\n",
      "Found 7 transitions\n",
      "Found 17 interactions\n",
      "  Converted to spreadsheet in 0.047s\n",
      "Found 7 species\n",
      "Found 7 transitions\n",
      "Found 17 interactions\n",
      "  Converted back to SBML in 0.076s\n",
      "\n",
      "[32/205] Processing: 209_ABERRANT-CELL-CYCLE-PROGRESSION_source.sbml\n",
      "Found 87 species\n",
      "Found 85 transitions\n",
      "Found 373 interactions\n",
      "  Converted to spreadsheet in 0.090s\n",
      "Found 87 species\n",
      "Found 85 transitions\n",
      "Found 373 interactions\n",
      "  Converted back to SBML in 0.163s\n",
      "\n",
      "[33/205] Processing: 077_SIGNALLING-PATHWAY-FOR-BUTANOL-PRODUCTION_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[34/205] Processing: 141_HIGH-OSMOLARITY-AND-MATING-PATHWAYS_source.sbml\n",
      "Found 45 species\n",
      "Found 43 transitions\n",
      "Found 94 interactions\n",
      "  Converted to spreadsheet in 0.056s\n",
      "Found 45 species\n",
      "Found 43 transitions\n",
      "Found 94 interactions\n",
      "  Converted back to SBML in 0.085s\n",
      "\n",
      "[35/205] Processing: 145_MELANOGENESIS_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; An SBML XML document must conform to the XML Schema for the corresponding SBML Level, Version and Release. The XML Schema for SBML defines the basic SBML object structure, the data types used by those objects, and the order in which the objects may appear in an SBML document.\n",
      " listOfInputs cannot be empty.\n",
      "\n",
      "\n",
      "[36/205] Processing: 238_DNA-DAMAGE-INDUCED-FERROPTOSIS_source.sbml\n",
      "Found 34 species\n",
      "Found 33 transitions\n",
      "Found 89 interactions\n",
      "  Converted to spreadsheet in 0.052s\n",
      "Found 34 species\n",
      "Found 33 transitions\n",
      "Found 89 interactions\n",
      "  Converted back to SBML in 0.087s\n",
      "\n",
      "[37/205] Processing: 104_DROSOPHILA-CELL-CYCLE_source.sbml\n",
      "Warning: Transition for target 'CycD' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Notch' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Ago' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 14 species\n",
      "Found 14 transitions\n",
      "Found 42 interactions\n",
      "  Converted to spreadsheet in 0.049s\n",
      "Found 14 species\n",
      "Found 14 transitions\n",
      "Found 42 interactions\n",
      "  Converted back to SBML in 0.082s\n",
      "\n",
      "[38/205] Processing: 184_P53-MDM2-NETWORK_source.sbml\n",
      "Warning: Transition for target 'Mdm2cyt_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 6 species\n",
      "Found 6 transitions\n",
      "Found 15 interactions\n",
      "  Converted to spreadsheet in 0.048s\n",
      "Found 6 species\n",
      "Found 6 transitions\n",
      "Found 15 interactions\n",
      "  Converted back to SBML in 0.073s\n",
      "\n",
      "[39/205] Processing: 018_EGFR-ERBB-SIGNALING_source.sbml\n",
      "Found 104 species\n",
      "Found 76 transitions\n",
      "Found 226 interactions\n",
      "  Converted to spreadsheet in 0.102s\n",
      "Found 104 species\n",
      "Found 76 transitions\n",
      "Found 226 interactions\n",
      "  Converted back to SBML in 0.120s\n",
      "\n",
      "[40/205] Processing: 176_MYELOFIBROTIC-MICROENVIRONMENT_source.sbml\n",
      "Found 49 species\n",
      "Found 47 transitions\n",
      "Found 148 interactions\n",
      "  Converted to spreadsheet in 0.069s\n",
      "Found 49 species\n",
      "Found 47 transitions\n",
      "Found 148 interactions\n",
      "  Converted back to SBML in 0.095s\n",
      "\n",
      "[41/205] Processing: 151_TCR-REDOX-METABOLISM_source.sbml\n",
      "Warning: Transition for target 'ETC_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 133 species\n",
      "Found 131 transitions\n",
      "Found 417 interactions\n",
      "  Converted to spreadsheet in 0.107s\n",
      "Found 133 species\n",
      "Found 131 transitions\n",
      "Found 417 interactions\n",
      "  Converted back to SBML in 0.286s\n",
      "\n",
      "[42/205] Processing: 132_VIRUS-REPLICATION-CYCLE_source.sbml\n",
      "Found 148 species\n",
      "Found 129 transitions\n",
      "Found 268 interactions\n",
      "  Converted to spreadsheet in 0.113s\n",
      "Found 148 species\n",
      "Found 129 transitions\n",
      "Found 268 interactions\n",
      "  Converted back to SBML in 0.126s\n",
      "\n",
      "[43/205] Processing: 046_B-BRONCHISEPTICA-AND-T-RETORTAEFORMIS_source.sbml\n",
      "Found 53 species\n",
      "Found 52 transitions\n",
      "Found 135 interactions\n",
      "  Converted to spreadsheet in 0.074s\n",
      "Found 53 species\n",
      "Found 52 transitions\n",
      "Found 135 interactions\n",
      "  Converted back to SBML in 0.113s\n",
      "\n",
      "[44/205] Processing: 016_IL-1-SIGNALING_source.sbml\n",
      "Found 118 species\n",
      "Found 104 transitions\n",
      "Found 218 interactions\n",
      "  Converted to spreadsheet in 0.117s\n",
      "Found 118 species\n",
      "Found 104 transitions\n",
      "Found 218 interactions\n",
      "  Converted back to SBML in 0.131s\n",
      "\n",
      "[45/205] Processing: 036_HCC1954-BREAST-CELL-LINE-SHORT-TERM_source.sbml\n",
      "Found 16 species\n",
      "Found 11 transitions\n",
      "Found 46 interactions\n",
      "  Converted to spreadsheet in 0.056s\n",
      "Found 16 species\n",
      "Found 11 transitions\n",
      "Found 46 interactions\n",
      "  Converted back to SBML in 0.087s\n",
      "\n",
      "[46/205] Processing: 040_T-CELL-DIFFERENTIATION_source.sbml\n",
      "Found 23 species\n",
      "Found 19 transitions\n",
      "Found 34 interactions\n",
      "  Converted to spreadsheet in 0.055s\n",
      "Found 23 species\n",
      "Found 19 transitions\n",
      "Found 34 interactions\n",
      "  Converted back to SBML in 0.085s\n",
      "\n",
      "[47/205] Processing: 283_EMT-MECHANOSENSING-TGFB_source.sbml\n",
      "Found 150 species\n",
      "Found 144 transitions\n",
      "Found 627 interactions\n",
      "  Converted to spreadsheet in 0.115s\n",
      "Found 150 species\n",
      "Found 144 transitions\n",
      "Found 627 interactions\n",
      "  Converted back to SBML in 0.268s\n",
      "\n",
      "[48/205] Processing: 191_SEGMENT-POLARITY-1-CELL_source.sbml\n",
      "Found 19 species\n",
      "Found 17 transitions\n",
      "Found 55 interactions\n",
      "  Converted to spreadsheet in 0.049s\n",
      "Found 19 species\n",
      "Found 17 transitions\n",
      "Found 55 interactions\n",
      "  Converted back to SBML in 0.082s\n",
      "\n",
      "[49/205] Processing: 122_NSP14_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[50/205] Processing: 200_LUNG-CANCER-CELL-CYCLE_source.sbml\n",
      "Found 26 species\n",
      "Found 25 transitions\n",
      "Found 71 interactions\n",
      "  Converted to spreadsheet in 0.055s\n",
      "Found 26 species\n",
      "Found 25 transitions\n",
      "Found 71 interactions\n",
      "  Converted back to SBML in 0.082s\n",
      "\n",
      "[51/205] Processing: 156_CONTROL-OF-TH1-TH2-TH17-TREG-DIFFERENTATION-REDUCED_source.sbml\n",
      "Found 36 species\n",
      "Found 23 transitions\n",
      "Found 108 interactions\n",
      "  Converted to spreadsheet in 0.053s\n",
      "Found 36 species\n",
      "Found 23 transitions\n",
      "Found 108 interactions\n",
      "  Converted back to SBML in 0.091s\n",
      "\n",
      "[52/205] Processing: 034_HCC1954-BREAST-CELL-LINE-LONG-TERM_source.sbml\n",
      "Found 23 species\n",
      "Found 19 transitions\n",
      "Found 68 interactions\n",
      "  Converted to spreadsheet in 0.067s\n",
      "Found 23 species\n",
      "Found 19 transitions\n",
      "Found 68 interactions\n",
      "  Converted back to SBML in 0.099s\n",
      "\n",
      "[53/205] Processing: 022_B-CELL-DIFFERENTIATION_source.sbml\n",
      "Found 22 species\n",
      "Found 17 transitions\n",
      "Found 39 interactions\n",
      "  Converted to spreadsheet in 0.058s\n",
      "Found 22 species\n",
      "Found 17 transitions\n",
      "Found 39 interactions\n",
      "  Converted back to SBML in 0.090s\n",
      "\n",
      "[54/205] Processing: 035_BT474-BREAST-CELL-LINE-SHORT-TERM_source.sbml\n",
      "Found 16 species\n",
      "Found 11 transitions\n",
      "Found 46 interactions\n",
      "  Converted to spreadsheet in 0.053s\n",
      "Found 16 species\n",
      "Found 11 transitions\n",
      "Found 46 interactions\n",
      "  Converted back to SBML in 0.087s\n",
      "\n",
      "[55/205] Processing: 082_TCR-TLR5-SIGNALING-2018_source.sbml\n",
      "Warning: Transition for target 'CD4' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CD45' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CD6' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'RASA1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'GADD45A' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Malt1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CCBL' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'PTPN22' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'RCAN1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Akap5' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Bcl10' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'RAC1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Kinases' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 128 species\n",
      "Found 125 transitions\n",
      "Found 257 interactions\n",
      "  Converted to spreadsheet in 0.084s\n",
      "Found 128 species\n",
      "Found 125 transitions\n",
      "Found 257 interactions\n",
      "  Converted back to SBML in 0.122s\n",
      "\n",
      "[56/205] Processing: 007_CORTICAL-AREA-DEVELOPMENT_source.sbml\n",
      "Found 5 species\n",
      "Found 5 transitions\n",
      "Found 14 interactions\n",
      "  Converted to spreadsheet in 0.047s\n",
      "Found 5 species\n",
      "Found 5 transitions\n",
      "Found 14 interactions\n",
      "  Converted back to SBML in 0.082s\n",
      "\n",
      "[57/205] Processing: 189_TRP-BIOSYNTHESIS_source.sbml\n",
      "Found 6 species\n",
      "Found 5 transitions\n",
      "Found 13 interactions\n",
      "  Converted to spreadsheet in 0.053s\n",
      "Found 6 species\n",
      "Found 5 transitions\n",
      "Found 13 interactions\n",
      "  Converted back to SBML in 0.087s\n",
      "\n",
      "[58/205] Processing: 274_MERGE-HEMATOPOIESIS-177-271_source.sbml\n",
      "Found 18 species\n",
      "Found 18 transitions\n",
      "Found 0 interactions\n",
      "  Converted to spreadsheet in 0.058s\n",
      "Found 18 species\n",
      "Found 18 transitions\n",
      "Found 0 interactions\n",
      "  Converted back to SBML in 0.107s\n",
      "\n",
      "[59/205] Processing: 070_MAPK-CANCER-CELL-FATE_source.sbml\n",
      "Found 53 species\n",
      "Found 49 transitions\n",
      "Found 104 interactions\n",
      "  Converted to spreadsheet in 0.082s\n",
      "Found 53 species\n",
      "Found 49 transitions\n",
      "Found 104 interactions\n",
      "  Converted back to SBML in 0.103s\n",
      "\n",
      "[60/205] Processing: 155_CONTROL-OF-TH1-TH2-TH17-TREG-DIFFERENTATION_source.sbml\n",
      "Warning: Transition for target 'IL15R_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IFNGR1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IFNGR2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'GP130' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL6RA' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL12RB1_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CGC' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL10RB' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL10RA' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL4RA_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL15RA' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL2RB' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL27RA' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 71 species\n",
      "Found 58 transitions\n",
      "Found 164 interactions\n",
      "  Converted to spreadsheet in 0.079s\n",
      "Found 71 species\n",
      "Found 58 transitions\n",
      "Found 164 interactions\n",
      "  Converted back to SBML in 0.118s\n",
      "\n",
      "[61/205] Processing: 061_TUMOR-MICROENVIRONMENT-IN-LYMPHOBLASTIC-LEUKEAMIA_source.sbml\n",
      "Found 26 species\n",
      "Found 24 transitions\n",
      "Found 79 interactions\n",
      "  Converted to spreadsheet in 0.078s\n",
      "Found 26 species\n",
      "Found 24 transitions\n",
      "Found 79 interactions\n",
      "  Converted back to SBML in 0.091s\n",
      "\n",
      "[62/205] Processing: 241_RHEUMATOID-ARTHRITIS-TH1_source.sbml\n",
      "Found 157 species\n",
      "Found 118 transitions\n",
      "Found 216 interactions\n",
      "  Converted to spreadsheet in 0.112s\n",
      "Found 157 species\n",
      "Found 118 transitions\n",
      "Found 216 interactions\n",
      "  Converted back to SBML in 0.118s\n",
      "\n",
      "[63/205] Processing: 139_ACUTE-RESPONSES-DURING-HYPERINSULINEMIA_source.sbml\n",
      "Found 19 species\n",
      "Found 10 transitions\n",
      "Found 0 interactions\n",
      "  Converted to spreadsheet in 0.056s\n",
      "Found 19 species\n",
      "Found 10 transitions\n",
      "Found 0 interactions\n",
      "  Converted back to SBML in 0.088s\n",
      "\n",
      "[64/205] Processing: 069_IRON-ACQUISITION-AND-STRESS-RESPONSE_source.sbml\n",
      "Found 22 species\n",
      "Found 20 transitions\n",
      "Found 38 interactions\n",
      "  Converted to spreadsheet in 0.056s\n",
      "Found 22 species\n",
      "Found 20 transitions\n",
      "Found 38 interactions\n",
      "  Converted back to SBML in 0.082s\n",
      "\n",
      "[65/205] Processing: 106_STOMATAL-RESTING-STATE_source.sbml\n",
      "Found 80 species\n",
      "Found 58 transitions\n",
      "Found 151 interactions\n",
      "  Converted to spreadsheet in 0.075s\n",
      "Found 80 species\n",
      "Found 58 transitions\n",
      "Found 151 interactions\n",
      "  Converted back to SBML in 0.103s\n",
      "\n",
      "[66/205] Processing: 159_BUDDING-YEAST-CORE_source.sbml\n",
      "Warning: Transition for target 'Bck2_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Clb5_b3' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Cdc20_b3' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Cdc15_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Cdc15_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Net1_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Cdc14_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Cdc14_b3' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 39 species\n",
      "Found 39 transitions\n",
      "Found 185 interactions\n",
      "  Converted to spreadsheet in 0.181s\n",
      "Found 39 species\n",
      "Found 39 transitions\n",
      "Found 185 interactions\n",
      "  Error converting to SBML: Unexpected end of expression. Missing closing parenthesis ')'. Found 294 opening '(' but only 293 closing ')' in: (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & Cln2 & !Swi5 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & !Swi5 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & !Swi5 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & !Swi5 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & !Swi5 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & !Swi5 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1\n",
      "\n",
      "[67/205] Processing: 095_FISSION-YEAST-2008_source.sbml\n",
      "Warning: Transition for target 'Start' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 10 species\n",
      "Found 10 transitions\n",
      "Found 27 interactions\n",
      "  Converted to spreadsheet in 0.050s\n",
      "Found 10 species\n",
      "Found 10 transitions\n",
      "Found 27 interactions\n",
      "  Converted back to SBML in 0.096s\n",
      "\n",
      "[68/205] Processing: 175_SEA-URCHIN_source.sbml\n",
      "Warning: Transition for target 'Panda' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Admp2_Trans' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 41 species\n",
      "Found 34 transitions\n",
      "Found 95 interactions\n",
      "  Converted to spreadsheet in 0.060s\n",
      "Found 41 species\n",
      "Found 34 transitions\n",
      "Found 95 interactions\n",
      "  Converted back to SBML in 0.098s\n",
      "\n",
      "[69/205] Processing: 063_LAC-OPERON_source.sbml\n",
      "Found 13 species\n",
      "Found 10 transitions\n",
      "Found 22 interactions\n",
      "  Converted to spreadsheet in 0.053s\n",
      "Found 13 species\n",
      "Found 10 transitions\n",
      "Found 22 interactions\n",
      "  Converted back to SBML in 0.082s\n",
      "\n",
      "[70/205] Processing: 013_CHOLESTEROL-REGULATORY-PATHWAY_source.sbml\n",
      "Found 34 species\n",
      "Found 32 transitions\n",
      "Found 41 interactions\n",
      "  Converted to spreadsheet in 0.058s\n",
      "Found 34 species\n",
      "Found 32 transitions\n",
      "Found 41 interactions\n",
      "  Converted back to SBML in 0.091s\n",
      "\n",
      "[71/205] Processing: 032_T-CELL-SIGNALLING-2006_source.sbml\n",
      "Found 40 species\n",
      "Found 37 transitions\n",
      "Found 53 interactions\n",
      "  Converted to spreadsheet in 0.081s\n",
      "Found 40 species\n",
      "Found 37 transitions\n",
      "Found 53 interactions\n",
      "  Converted back to SBML in 0.096s\n",
      "\n",
      "[72/205] Processing: 097_DROSOPHILA-WINGS-AP_source.sbml\n",
      "Warning: Transition for target 'Ap' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Ap2' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 10 species\n",
      "Found 10 transitions\n",
      "Found 14 interactions\n",
      "  Converted to spreadsheet in 0.053s\n",
      "Found 10 species\n",
      "Found 10 transitions\n",
      "Found 14 interactions\n",
      "  Converted back to SBML in 0.092s\n",
      "\n",
      "[73/205] Processing: 214_FOXO3-PATHWAY_source.sbml\n",
      "Found 70 species\n",
      "Found 52 transitions\n",
      "Found 89 interactions\n",
      "  Converted to spreadsheet in 0.086s\n",
      "Found 70 species\n",
      "Found 52 transitions\n",
      "Found 89 interactions\n",
      "  Converted back to SBML in 0.095s\n",
      "\n",
      "[74/205] Processing: 165_EGGSHELL-PATTERNING-PHENOMOENOLOGICAL_source.sbml\n",
      "Found 8 species\n",
      "Found 4 transitions\n",
      "Found 16 interactions\n",
      "  Converted to spreadsheet in 0.047s\n",
      "Found 8 species\n",
      "Found 4 transitions\n",
      "Found 16 interactions\n",
      "  Converted back to SBML in 0.078s\n",
      "\n",
      "[75/205] Processing: 187_MAMMAL-SEX-DETERMINATION-1-CELL_source.sbml\n",
      "Found 19 species\n",
      "Found 13 transitions\n",
      "Found 55 interactions\n",
      "  Converted to spreadsheet in 0.054s\n",
      "Found 19 species\n",
      "Found 13 transitions\n",
      "Found 55 interactions\n",
      "  Converted back to SBML in 0.081s\n",
      "\n",
      "[76/205] Processing: 136_EGF-TNF-ALPHA-SIGNALLING-PATHWAY_source.sbml\n",
      "Found 28 species\n",
      "Found 26 transitions\n",
      "Found 31 interactions\n",
      "  Converted to spreadsheet in 0.051s\n",
      "Found 28 species\n",
      "Found 26 transitions\n",
      "Found 31 interactions\n",
      "  Converted back to SBML in 0.084s\n",
      "\n",
      "[77/205] Processing: 131_TGFB-PATHWAY_source.sbml\n",
      "Found 21 species\n",
      "Found 7 transitions\n",
      "Found 24 interactions\n",
      "  Converted to spreadsheet in 0.047s\n",
      "Found 21 species\n",
      "Found 7 transitions\n",
      "Found 24 interactions\n",
      "  Converted back to SBML in 0.078s\n",
      "\n",
      "[78/205] Processing: 015_NEUROTRANSMITTER-SIGNALING-PATHWAY_source.sbml\n",
      "Found 16 species\n",
      "Found 14 transitions\n",
      "Found 20 interactions\n",
      "  Converted to spreadsheet in 0.052s\n",
      "Found 16 species\n",
      "Found 14 transitions\n",
      "Found 20 interactions\n",
      "  Converted back to SBML in 0.078s\n",
      "\n",
      "[79/205] Processing: 216_PPARGC1A-PATHWAY_source.sbml\n",
      "Found 69 species\n",
      "Found 52 transitions\n",
      "Found 109 interactions\n",
      "  Converted to spreadsheet in 0.089s\n",
      "Found 69 species\n",
      "Found 52 transitions\n",
      "Found 109 interactions\n",
      "  Converted back to SBML in 0.097s\n",
      "\n",
      "[80/205] Processing: 219_WNT-PI3K-AKT-PATHWAY_source.sbml\n",
      "Found 84 species\n",
      "Found 45 transitions\n",
      "Found 106 interactions\n",
      "  Converted to spreadsheet in 0.089s\n",
      "Found 84 species\n",
      "Found 45 transitions\n",
      "Found 106 interactions\n",
      "  Converted back to SBML in 0.100s\n",
      "\n",
      "[81/205] Processing: 135_SIGNAL-TRANSDUCTION_source.sbml\n",
      "Found 30 species\n",
      "Found 28 transitions\n",
      "Found 33 interactions\n",
      "  Converted to spreadsheet in 0.047s\n",
      "Found 30 species\n",
      "Found 28 transitions\n",
      "Found 33 interactions\n",
      "  Converted back to SBML in 0.083s\n",
      "\n",
      "[82/205] Processing: 203_DDR-SIGNALLING-PATHWAYS_source.sbml\n",
      "Found 40 species\n",
      "Found 39 transitions\n",
      "Found 143 interactions\n",
      "  Converted to spreadsheet in 0.067s\n",
      "Found 40 species\n",
      "Found 39 transitions\n",
      "Found 143 interactions\n",
      "  Converted back to SBML in 0.091s\n",
      "\n",
      "[83/205] Processing: 088_MIR-9-NEUROGENESIS_source.sbml\n",
      "Found 6 species\n",
      "Found 6 transitions\n",
      "Found 11 interactions\n",
      "  Converted to spreadsheet in 0.049s\n",
      "Warning: Species 'Her6' (row 2): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'HuC' (row 3): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Zic5' (row 6): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'miR9' (row 7): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Found 6 species\n",
      "Found 6 transitions\n",
      "Found 11 interactions\n",
      "  Converted back to SBML in 0.073s\n",
      "\n",
      "[84/205] Processing: 277_HYPOXIA-AND-MECHANOSENSING-IN-METASTASIS_source.sbml\n",
      "Found 165 species\n",
      "Found 158 transitions\n",
      "Found 755 interactions\n",
      "  Converted to spreadsheet in 0.132s\n",
      "Found 165 species\n",
      "Found 158 transitions\n",
      "Found 755 interactions\n",
      "  Converted back to SBML in 0.411s\n",
      "\n",
      "[85/205] Processing: 108_GEROCONVERSION_source.sbml\n",
      "Found 25 species\n",
      "Found 23 transitions\n",
      "Found 67 interactions\n",
      "  Converted to spreadsheet in 0.067s\n",
      "Found 25 species\n",
      "Found 23 transitions\n",
      "Found 67 interactions\n",
      "  Converted back to SBML in 0.079s\n",
      "\n",
      "[86/205] Processing: 196_T-LYMPHOCYTE-SPECIFICATION_source.sbml\n",
      "Found 61 species\n",
      "Found 58 transitions\n",
      "Found 237 interactions\n",
      "  Converted to spreadsheet in 0.076s\n",
      "Found 61 species\n",
      "Found 58 transitions\n",
      "Found 237 interactions\n",
      "  Converted back to SBML in 0.107s\n",
      "\n",
      "[87/205] Processing: 003_MAMMALIAN-CELL-CYCLE_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[88/205] Processing: 062_CD4-T-CELL-DIFFERENTIATION-AND-PLASTICITY_source.sbml\n",
      "Found 18 species\n",
      "Found 12 transitions\n",
      "Found 78 interactions\n",
      "  Converted to spreadsheet in 0.051s\n",
      "Found 18 species\n",
      "Found 12 transitions\n",
      "Found 78 interactions\n",
      "  Converted back to SBML in 0.080s\n",
      "\n",
      "[89/205] Processing: 042_TOL-REGULATORY-NETWORK_source.sbml\n",
      "Found 24 species\n",
      "Found 14 transitions\n",
      "Found 48 interactions\n",
      "  Converted to spreadsheet in 0.059s\n",
      "Found 24 species\n",
      "Found 14 transitions\n",
      "Found 48 interactions\n",
      "  Converted back to SBML in 0.115s\n",
      "\n",
      "[90/205] Processing: 099_YEAST-HYPHAL-TRANSITION_source.sbml\n",
      "Found 19 species\n",
      "Found 16 transitions\n",
      "Found 36 interactions\n",
      "  Converted to spreadsheet in 0.051s\n",
      "Found 19 species\n",
      "Found 16 transitions\n",
      "Found 36 interactions\n",
      "  Converted back to SBML in 0.077s\n",
      "\n",
      "[91/205] Processing: 120_KYNURENINE-PATHWAY_source.sbml\n",
      "Found 150 species\n",
      "Found 78 transitions\n",
      "Found 304 interactions\n",
      "  Converted to spreadsheet in 0.103s\n",
      "Found 150 species\n",
      "Found 78 transitions\n",
      "Found 304 interactions\n",
      "  Converted back to SBML in 0.120s\n",
      "\n",
      "[92/205] Processing: 071_CASTRATION-RESISTANT-PROSTATE-CANCER_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "\n",
      "\n",
      "[93/205] Processing: 005_FA-BRCA-PATHWAY_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[94/205] Processing: 021_BODY-SEGMENTATION-IN-DROSOPHILA-2013_source.sbml\n",
      "Found 17 species\n",
      "Found 14 transitions\n",
      "Found 29 interactions\n",
      "  Converted to spreadsheet in 0.048s\n",
      "Found 17 species\n",
      "Found 14 transitions\n",
      "Found 29 interactions\n",
      "  Converted back to SBML in 0.080s\n",
      "\n",
      "[95/205] Processing: 164_EGGSHELL-PATTERNING-MECHANISTIC_source.sbml\n",
      "Found 24 species\n",
      "Found 17 transitions\n",
      "Found 62 interactions\n",
      "  Converted to spreadsheet in 0.058s\n",
      "Found 24 species\n",
      "Found 17 transitions\n",
      "Found 62 interactions\n",
      "  Converted back to SBML in 0.098s\n",
      "\n",
      "[96/205] Processing: 195_CTLA4-PD1-CHECKPOINT-INHIBITORS_source.sbml\n",
      "Warning: Transition for target 'i_cAMP' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_MAL' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_UNC119' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_RAB11A' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_PTPRJ' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_14_3_3' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_PRKDC' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_MDM2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_NCKAP1L' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_NCKAP1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_BAIAP2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_CYFIP1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_BRK1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_WASF2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_ABI1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_ABI2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_ACTR2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_ACTR3' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_PRDX1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_CRACR2A' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_ROS' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_RCAN1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_CABIN1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_BBC3' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_APAF1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_DEPTOR' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_RICTOR' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_MLST8' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_MTOR' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_Glucose' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_DUSP2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_DUSP4' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'DUSP5' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_DUSP6' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_DUSP10' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_DUSP22' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_Axin1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_SMAD3' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_Th1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'f_BAT3' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'INPP5D' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 216 species\n",
      "Found 202 transitions\n",
      "Found 439 interactions\n",
      "  Converted to spreadsheet in 0.107s\n",
      "Found 216 species\n",
      "Found 202 transitions\n",
      "Found 439 interactions\n",
      "  Converted back to SBML in 0.243s\n",
      "\n",
      "[97/205] Processing: 019_IL-6-SIGNALING_source.sbml\n",
      "Found 86 species\n",
      "Found 71 transitions\n",
      "Found 149 interactions\n",
      "  Converted to spreadsheet in 0.097s\n",
      "Found 86 species\n",
      "Found 71 transitions\n",
      "Found 149 interactions\n",
      "  Converted back to SBML in 0.113s\n",
      "\n",
      "[98/205] Processing: 166_DROSOPHILA-JAK-STAT-PATHWAY_source.sbml\n",
      "Found 19 species\n",
      "Found 7 transitions\n",
      "Found 36 interactions\n",
      "  Converted to spreadsheet in 0.048s\n",
      "Found 19 species\n",
      "Found 7 transitions\n",
      "Found 36 interactions\n",
      "  Converted back to SBML in 0.084s\n",
      "\n",
      "[99/205] Processing: 059_BORTEZOMIB-RESPONSES-IN-MYELOMA-CELLS_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[100/205] Processing: 181_MULTILEVEL-CELL-CYCLE_source.sbml\n",
      "Found 13 species\n",
      "Found 12 transitions\n",
      "Found 59 interactions\n",
      "  Converted to spreadsheet in 0.053s\n",
      "Found 13 species\n",
      "Found 12 transitions\n",
      "Found 59 interactions\n",
      "  Converted back to SBML in 0.083s\n",
      "\n",
      "[101/205] Processing: 008_DEATH-RECEPTOR-SIGNALING_source.sbml\n",
      "Found 28 species\n",
      "Found 25 transitions\n",
      "Found 45 interactions\n",
      "  Converted to spreadsheet in 0.054s\n",
      "Found 28 species\n",
      "Found 25 transitions\n",
      "Found 45 interactions\n",
      "  Converted back to SBML in 0.087s\n",
      "\n",
      "[102/205] Processing: 272_FLT3-ITD-AML-SIGNALING_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[103/205] Processing: 130_RENIN-ANGIOTENSIN_source.sbml\n",
      "Found 77 species\n",
      "Found 43 transitions\n",
      "Found 130 interactions\n",
      "  Converted to spreadsheet in 0.244s\n",
      "Found 77 species\n",
      "Found 43 transitions\n",
      "Found 130 interactions\n",
      "  Converted back to SBML in 0.173s\n",
      "\n",
      "[104/205] Processing: 148_AGS-CELL-FATE-DECISION_source.sbml\n",
      "Found 83 species\n",
      "Found 83 transitions\n",
      "Found 185 interactions\n",
      "  Converted to spreadsheet in 0.081s\n",
      "Found 83 species\n",
      "Found 83 transitions\n",
      "Found 185 interactions\n",
      "  Converted back to SBML in 0.108s\n",
      "\n",
      "[105/205] Processing: 109_ASYMMETRIC-CELL-DIVISION-A_source.sbml\n",
      "Found 5 species\n",
      "Found 5 transitions\n",
      "Found 15 interactions\n",
      "  Converted to spreadsheet in 0.050s\n",
      "Found 5 species\n",
      "Found 5 transitions\n",
      "Found 15 interactions\n",
      "  Converted back to SBML in 0.087s\n",
      "\n",
      "[106/205] Processing: 014_T-LGL-SURVIVAL-NETWORK-2008_source.sbml\n",
      "Found 61 species\n",
      "Found 54 transitions\n",
      "Found 193 interactions\n",
      "  Converted to spreadsheet in 0.095s\n",
      "Found 61 species\n",
      "Found 54 transitions\n",
      "Found 193 interactions\n",
      "  Converted back to SBML in 0.113s\n",
      "\n",
      "[107/205] Processing: 068_AURORA-KINASE-A-IN-NEUROBLASTOMA_source.sbml\n",
      "Found 23 species\n",
      "Found 19 transitions\n",
      "Found 43 interactions\n",
      "  Converted to spreadsheet in 0.058s\n",
      "Found 23 species\n",
      "Found 19 transitions\n",
      "Found 43 interactions\n",
      "  Converted back to SBML in 0.083s\n",
      "\n",
      "[108/205] Processing: 170_DROSOPHILA-GAP-B_source.sbml\n",
      "Warning: Transition for target 'Hb_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Hb_b3' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Kni' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 7 species\n",
      "Found 7 transitions\n",
      "Found 15 interactions\n",
      "  Converted to spreadsheet in 0.047s\n",
      "Found 7 species\n",
      "Found 7 transitions\n",
      "Found 15 interactions\n",
      "  Converted back to SBML in 0.073s\n",
      "\n",
      "[109/205] Processing: 282_EMT-MECHANOSENSING_source.sbml\n",
      "Found 136 species\n",
      "Found 132 transitions\n",
      "Found 567 interactions\n",
      "  Converted to spreadsheet in 0.114s\n",
      "Found 136 species\n",
      "Found 132 transitions\n",
      "Found 567 interactions\n",
      "  Converted back to SBML in 0.234s\n",
      "\n",
      "[110/205] Processing: 134_RHEUMATOID-ARTHRITIS_source.sbml\n",
      "Found 38 species\n",
      "Found 35 transitions\n",
      "Found 59 interactions\n",
      "  Converted to spreadsheet in 0.054s\n",
      "Found 38 species\n",
      "Found 35 transitions\n",
      "Found 59 interactions\n",
      "  Converted back to SBML in 0.085s\n",
      "\n",
      "[111/205] Processing: 152_TCR-REDOX-METABOLISM-REDUCED_source.sbml\n",
      "Warning: Transition for target 'ETC_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 53 species\n",
      "Found 51 transitions\n",
      "Found 288 interactions\n",
      "  Converted to spreadsheet in 0.137s\n",
      "Found 53 species\n",
      "Found 51 transitions\n",
      "Found 288 interactions\n",
      "  Converted back to SBML in 0.632s\n",
      "\n",
      "[112/205] Processing: 186_CHICKEN-SEX-DETERMINATION-REDUCED_source.sbml\n",
      "Found 10 species\n",
      "Found 7 transitions\n",
      "Found 25 interactions\n",
      "  Converted to spreadsheet in 0.051s\n",
      "Found 10 species\n",
      "Found 7 transitions\n",
      "Found 25 interactions\n",
      "  Converted back to SBML in 0.086s\n",
      "\n",
      "[113/205] Processing: 004_ERBB-RECEPTOR-SIGNALING_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[114/205] Processing: 086_TUMOUR-CELL-INVASION-AND-MIGRATION-REDUCED_source.sbml\n",
      "Found 20 species\n",
      "Found 18 transitions\n",
      "Found 88 interactions\n",
      "  Converted to spreadsheet in 0.055s\n",
      "Found 20 species\n",
      "Found 18 transitions\n",
      "Found 88 interactions\n",
      "  Converted back to SBML in 0.096s\n",
      "\n",
      "[115/205] Processing: 157_CONTROL-OF-TH-DIFFERENTATION_source.sbml\n",
      "Warning: Transition for target 'IFNAR1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IFNAR2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IFNGR1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL1RAP' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IFNGR2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'GP130' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL6RA' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL12RB1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CGC' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL10RB' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL10RA' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL4RA' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL15RA' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL2RB' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL27RA' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL17RB' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL18RAP' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL18RA' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL1RL2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL28RA' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 103 species\n",
      "Found 82 transitions\n",
      "Found 234 interactions\n",
      "  Converted to spreadsheet in 0.082s\n",
      "Found 103 species\n",
      "Found 82 transitions\n",
      "Found 234 interactions\n",
      "  Converted back to SBML in 0.122s\n",
      "\n",
      "[116/205] Processing: 188_MAMMAL-SEX-DETERMINATION-2-CELL_source.sbml\n",
      "Found 37 species\n",
      "Found 30 transitions\n",
      "Found 132 interactions\n",
      "  Converted to spreadsheet in 0.074s\n",
      "Found 37 species\n",
      "Found 30 transitions\n",
      "Found 132 interactions\n",
      "  Converted back to SBML in 0.095s\n",
      "\n",
      "[117/205] Processing: 064_METABOLIC-INTERACTIONS-IN-GUT-MICROBIOME_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "\n",
      "\n",
      "[118/205] Processing: 057_FANCONI-ANEMIA-AND-CHECKPOINT-RECOVERY_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[119/205] Processing: 044_TRICHOSTRONGYLUS-RETORTAEFORMIS_source.sbml\n",
      "Found 26 species\n",
      "Found 25 transitions\n",
      "Found 58 interactions\n",
      "  Converted to spreadsheet in 0.058s\n",
      "Found 26 species\n",
      "Found 25 transitions\n",
      "Found 58 interactions\n",
      "  Converted back to SBML in 0.095s\n",
      "\n",
      "[120/205] Processing: 031_CELL-CYCLE-TRANSCRIPTION_source.sbml\n",
      "Found 9 species\n",
      "Found 9 transitions\n",
      "Found 19 interactions\n",
      "  Converted to spreadsheet in 0.053s\n",
      "Found 9 species\n",
      "Found 9 transitions\n",
      "Found 19 interactions\n",
      "  Converted back to SBML in 0.083s\n",
      "\n",
      "[121/205] Processing: 167_MESODERM-SPECIFICATION-IN-DROSOPHILA_source.sbml\n",
      "Warning: Transition for target 'Bin_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Pnt_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Ras_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Rl_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 57 species\n",
      "Found 45 transitions\n",
      "Found 130 interactions\n",
      "  Converted to spreadsheet in 0.075s\n",
      "Found 57 species\n",
      "Found 45 transitions\n",
      "Found 130 interactions\n",
      "  Converted back to SBML in 0.112s\n",
      "\n",
      "[122/205] Processing: 194_VULVAR-PRECURSOR-CELLS_source.sbml\n",
      "Warning: Transition for target 'AEGL20_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'AEGL20_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'ACWN1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'ACWN2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'ALIN44' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'AMOM2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CEH13' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CEH20' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'UNC62' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'LIN31' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'LIN1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'KSR2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'KSR1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'LIN10' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'LIN7' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'LIN2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'LIT1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'VANG1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'LAG1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'SEL8' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'PEN2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'APH1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'SUP17' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'lin4' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 106 species\n",
      "Found 102 transitions\n",
      "Found 236 interactions\n",
      "  Converted to spreadsheet in 0.085s\n",
      "Found 106 species\n",
      "Found 102 transitions\n",
      "Found 236 interactions\n",
      "  Converted back to SBML in 0.114s\n",
      "\n",
      "[123/205] Processing: 185_CHICKEN-SEX-DETERMINATION_source.sbml\n",
      "Found 15 species\n",
      "Found 12 transitions\n",
      "Found 34 interactions\n",
      "  Converted to spreadsheet in 0.054s\n",
      "Found 15 species\n",
      "Found 12 transitions\n",
      "Found 34 interactions\n",
      "  Converted back to SBML in 0.093s\n",
      "\n",
      "[124/205] Processing: 163_DROSOPHILA-EGF-PATHWAY_source.sbml\n",
      "Found 34 species\n",
      "Found 24 transitions\n",
      "Found 84 interactions\n",
      "  Converted to spreadsheet in 0.055s\n",
      "Found 34 species\n",
      "Found 24 transitions\n",
      "Found 84 interactions\n",
      "  Converted back to SBML in 0.098s\n",
      "\n",
      "[125/205] Processing: 142_BLOOD-STEM-CELL_source.sbml\n",
      "Found 29 species\n",
      "Found 27 transitions\n",
      "Found 0 interactions\n",
      "  Converted to spreadsheet in 0.058s\n",
      "Found 29 species\n",
      "Found 27 transitions\n",
      "Found 0 interactions\n",
      "  Converted back to SBML in 0.102s\n",
      "\n",
      "[126/205] Processing: 149_AGS-CELL-FATE-DECISION-REDUCED_source.sbml\n",
      "Found 14 species\n",
      "Found 12 transitions\n",
      "Found 62 interactions\n",
      "  Converted to spreadsheet in 0.050s\n",
      "Found 14 species\n",
      "Found 12 transitions\n",
      "Found 62 interactions\n",
      "  Converted back to SBML in 0.084s\n",
      "\n",
      "[127/205] Processing: 058_ARABIDOPSIS-THALIANA-CELL-CYCLE_source.sbml\n",
      "Found 14 species\n",
      "Found 14 transitions\n",
      "Found 66 interactions\n",
      "  Converted to spreadsheet in 0.053s\n",
      "Found 14 species\n",
      "Found 14 transitions\n",
      "Found 66 interactions\n",
      "  Converted back to SBML in 0.085s\n",
      "\n",
      "[128/205] Processing: 143_BREAST-CANCER-INHIBITORS_source.sbml\n",
      "Found 97 species\n",
      "Found 71 transitions\n",
      "Found 223 interactions\n",
      "  Converted to spreadsheet in 0.075s\n",
      "Found 97 species\n",
      "Found 71 transitions\n",
      "Found 223 interactions\n",
      "  Converted back to SBML in 0.120s\n",
      "\n",
      "[129/205] Processing: 146_BUDDING-YEAST-FAURE-2009_source.sbml\n",
      "Warning: Transition for target 'Bck2_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Clb5_b3' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Cdc20_b3' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'PP2ACdc55_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Cdc15_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Net1_b3' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Cdc14_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Cdc14_b3' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'SeparaseEsp1_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'SeparaseEsp1_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 50 species\n",
      "Found 50 transitions\n",
      "Found 271 interactions\n",
      "  Converted to spreadsheet in 0.219s\n",
      "Found 50 species\n",
      "Found 50 transitions\n",
      "Found 271 interactions\n",
      "  Error converting to SBML: Unexpected end of expression. Missing closing parenthesis ')'. Found 277 opening '(' but only 276 closing ')' in: (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & Cln2 & !Swi5 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & !Swi5 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & !Swi5 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & !Swi5 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & !Swi5 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & !Swi5 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & Bck2_b1\n",
      "\n",
      "[130/205] Processing: 133_ROOT-STEM-CELL-2010_source.sbml\n",
      "Found 10 species\n",
      "Found 8 transitions\n",
      "Found 0 interactions\n",
      "  Converted to spreadsheet in 0.045s\n",
      "Found 10 species\n",
      "Found 8 transitions\n",
      "Found 0 interactions\n",
      "  Converted back to SBML in 0.076s\n",
      "\n",
      "[131/205] Processing: 055_HUMAN-GONADAL-SEX-DETERMINATION_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[132/205] Processing: 213_DOPAMINE-TRANSCRIPTION_source.sbml\n",
      "Found 65 species\n",
      "Found 54 transitions\n",
      "Found 94 interactions\n",
      "  Converted to spreadsheet in 0.093s\n",
      "Found 65 species\n",
      "Found 54 transitions\n",
      "Found 94 interactions\n",
      "  Converted back to SBML in 0.101s\n",
      "\n",
      "[133/205] Processing: 085_REPROGRAMMING-TESTES-DERIVED-STEM-CELLS_source.sbml\n",
      "Found 27 species\n",
      "Found 23 transitions\n",
      "Found 49 interactions\n",
      "  Converted to spreadsheet in 0.050s\n",
      "Found 27 species\n",
      "Found 23 transitions\n",
      "Found 49 interactions\n",
      "  Converted back to SBML in 0.089s\n",
      "\n",
      "[134/205] Processing: 096_ERBB-REGULATED-G1-S-TRANSITION_source.sbml\n",
      "Warning: Transition for target 'EGF' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 20 species\n",
      "Found 20 transitions\n",
      "Found 48 interactions\n",
      "  Converted to spreadsheet in 0.052s\n",
      "Found 20 species\n",
      "Found 20 transitions\n",
      "Found 48 interactions\n",
      "  Converted back to SBML in 0.091s\n",
      "\n",
      "[135/205] Processing: 111_APOPTOSIS_source.sbml\n",
      "Found 33 species\n",
      "Found 18 transitions\n",
      "Found 40 interactions\n",
      "  Converted to spreadsheet in 0.063s\n",
      "Found 33 species\n",
      "Found 18 transitions\n",
      "Found 40 interactions\n",
      "  Converted back to SBML in 0.094s\n",
      "\n",
      "[136/205] Processing: 017_DIFFERENTIATION-OF-T-LYMPHOCYTES_source.sbml\n",
      "Found 50 species\n",
      "Found 41 transitions\n",
      "Found 97 interactions\n",
      "  Converted to spreadsheet in 0.076s\n",
      "Found 50 species\n",
      "Found 41 transitions\n",
      "Found 97 interactions\n",
      "  Converted back to SBML in 0.109s\n",
      "\n",
      "[137/205] Processing: 037_SKBR3-BREAST-CELL-LINE-SHORT-TERM_source.sbml\n",
      "Found 16 species\n",
      "Found 11 transitions\n",
      "Found 41 interactions\n",
      "  Converted to spreadsheet in 0.071s\n",
      "Found 16 species\n",
      "Found 11 transitions\n",
      "Found 41 interactions\n",
      "  Converted back to SBML in 0.099s\n",
      "\n",
      "[138/205] Processing: 110_ASYMMETRIC-CELL-DIVISION-B_source.sbml\n",
      "Found 9 species\n",
      "Found 9 transitions\n",
      "Found 12 interactions\n",
      "  Converted to spreadsheet in 0.052s\n",
      "Found 9 species\n",
      "Found 9 transitions\n",
      "Found 12 interactions\n",
      "  Converted back to SBML in 0.089s\n",
      "\n",
      "[139/205] Processing: 039_HIV-1-INTERACTIONS-WITH-T-CELL-SIGNALING_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[140/205] Processing: 116_HMOX1-PATHWAY_source.sbml\n",
      "Found 144 species\n",
      "Found 89 transitions\n",
      "Found 228 interactions\n",
      "  Converted to spreadsheet in 0.111s\n",
      "Found 144 species\n",
      "Found 89 transitions\n",
      "Found 228 interactions\n",
      "  Converted back to SBML in 0.126s\n",
      "\n",
      "[141/205] Processing: 006_HGF-SIGNALING-IN-KERATINOCYTES_source.sbml\n",
      "Found 68 species\n",
      "Found 62 transitions\n",
      "Found 103 interactions\n",
      "  Converted to spreadsheet in 0.088s\n",
      "Found 68 species\n",
      "Found 62 transitions\n",
      "Found 103 interactions\n",
      "  Converted back to SBML in 0.109s\n",
      "\n",
      "[142/205] Processing: 162_DROSOPHILA-DPP-PATHWAY_source.sbml\n",
      "Found 18 species\n",
      "Found 10 transitions\n",
      "Found 40 interactions\n",
      "  Converted to spreadsheet in 0.055s\n",
      "Found 18 species\n",
      "Found 10 transitions\n",
      "Found 40 interactions\n",
      "  Converted back to SBML in 0.090s\n",
      "\n",
      "[143/205] Processing: 045_HH-PATHWAY-OF-DROSOPHILA_source.sbml\n",
      "Found 24 species\n",
      "Found 11 transitions\n",
      "Found 32 interactions\n",
      "  Converted to spreadsheet in 0.059s\n",
      "Found 24 species\n",
      "Found 11 transitions\n",
      "Found 32 interactions\n",
      "  Converted back to SBML in 0.080s\n",
      "\n",
      "[144/205] Processing: 284_ANCHORAGE-DEPENDENCE-PROLIFERATION_source.sbml\n",
      "Found 121 species\n",
      "Found 117 transitions\n",
      "Found 491 interactions\n",
      "  Converted to spreadsheet in 0.108s\n",
      "Found 121 species\n",
      "Found 117 transitions\n",
      "Found 491 interactions\n",
      "  Converted back to SBML in 0.225s\n",
      "\n",
      "[145/205] Processing: 081_TLR5-SIGNALING-2018_source.sbml\n",
      "Warning: Transition for target 'SCFbTrCP' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 42 species\n",
      "Found 41 transitions\n",
      "Found 68 interactions\n",
      "  Converted to spreadsheet in 0.057s\n",
      "Found 42 species\n",
      "Found 41 transitions\n",
      "Found 68 interactions\n",
      "  Converted back to SBML in 0.100s\n",
      "\n",
      "[146/205] Processing: 049_OXIDATIVE-STRESS-PATHWAY_source.sbml\n",
      "Found 19 species\n",
      "Found 18 transitions\n",
      "Found 32 interactions\n",
      "  Converted to spreadsheet in 0.132s\n",
      "Found 19 species\n",
      "Found 18 transitions\n",
      "Found 32 interactions\n",
      "  Converted back to SBML in 0.092s\n",
      "\n",
      "[147/205] Processing: 147_BUDDING-YEAST-EXIT-MODULE_source.sbml\n",
      "Warning: Transition for target 'SeparaseEsp1_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'SeparaseEsp1_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'PP2ACdc55_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Tem1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Cdc14' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 16 species\n",
      "Found 16 transitions\n",
      "Found 44 interactions\n",
      "  Converted to spreadsheet in 0.050s\n",
      "Found 16 species\n",
      "Found 16 transitions\n",
      "Found 44 interactions\n",
      "  Converted back to SBML in 0.095s\n",
      "\n",
      "[148/205] Processing: 150_CELL-FATE-DECISION-MULTISCALE_source.sbml\n",
      "Found 33 species\n",
      "Found 31 transitions\n",
      "Found 52 interactions\n",
      "  Converted to spreadsheet in 0.059s\n",
      "Warning: Species 'ATP' (row 2): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'BAX' (row 4): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'BCL2' (row 5): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CASP3' (row 6): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CASP8' (row 7): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Cyt_c' (row 8): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'DISC_TNF' (row 11): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'FASL' (row 14): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'IKK' (row 15): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'NFKB1' (row 18): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'NonACD' (row 19): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'RIP1' (row 20): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'RIP1ub' (row 22): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'ROS' (row 23): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'SMAC' (row 24): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'TNFR' (row 27): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'cFLIP' (row 30): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Found 33 species\n",
      "Found 31 transitions\n",
      "Found 52 interactions\n",
      "  Converted back to SBML in 0.099s\n",
      "\n",
      "[149/205] Processing: 125_ORF10-CUL2-PATHWAY_source.sbml\n",
      "Found 51 species\n",
      "Found 34 transitions\n",
      "Found 92 interactions\n",
      "  Converted to spreadsheet in 0.083s\n",
      "Found 51 species\n",
      "Found 34 transitions\n",
      "Found 92 interactions\n",
      "  Converted back to SBML in 0.096s\n",
      "\n",
      "[150/205] Processing: 128_PYRIMIDINE-DEPRIVATION_source.sbml\n",
      "Found 90 species\n",
      "Found 56 transitions\n",
      "Found 131 interactions\n",
      "  Converted to spreadsheet in 0.094s\n",
      "Found 90 species\n",
      "Found 56 transitions\n",
      "Found 131 interactions\n",
      "  Converted back to SBML in 0.108s\n",
      "\n",
      "[151/205] Processing: 113_ER-STRESS_source.sbml\n",
      "Found 182 species\n",
      "Found 107 transitions\n",
      "Found 266 interactions\n",
      "  Converted to spreadsheet in 0.128s\n",
      "Found 182 species\n",
      "Found 107 transitions\n",
      "Found 266 interactions\n",
      "  Converted back to SBML in 0.131s\n",
      "\n",
      "[152/205] Processing: 280_SHH-SIGNALING-EMT-REGULATION_source.sbml\n",
      "Found 176 species\n",
      "Found 169 transitions\n",
      "Found 781 interactions\n",
      "  Converted to spreadsheet in 0.134s\n",
      "Found 176 species\n",
      "Found 169 transitions\n",
      "Found 781 interactions\n",
      "  Converted back to SBML in 0.404s\n",
      "\n",
      "[153/205] Processing: 023_MAMMALIAN-CELL-CYCLE-2006_source.sbml\n",
      "Found 10 species\n",
      "Found 9 transitions\n",
      "Found 34 interactions\n",
      "  Converted to spreadsheet in 0.057s\n",
      "Found 10 species\n",
      "Found 9 transitions\n",
      "Found 34 interactions\n",
      "  Converted back to SBML in 0.092s\n",
      "\n",
      "[154/205] Processing: 115_E-PROTEIN_source.sbml\n",
      "Found 35 species\n",
      "Found 17 transitions\n",
      "Found 40 interactions\n",
      "  Converted to spreadsheet in 0.062s\n",
      "Found 35 species\n",
      "Found 17 transitions\n",
      "Found 40 interactions\n",
      "  Converted back to SBML in 0.099s\n",
      "\n",
      "[155/205] Processing: 025_T-LGL-SURVIVAL-NETWORK-2011_source.sbml\n",
      "Found 60 species\n",
      "Found 54 transitions\n",
      "Found 195 interactions\n",
      "  Converted to spreadsheet in 0.101s\n",
      "Found 60 species\n",
      "Found 54 transitions\n",
      "Found 195 interactions\n",
      "  Converted back to SBML in 0.115s\n",
      "\n",
      "[156/205] Processing: 137_SIGNALLING-IN-LIVER-CANCER_source.sbml\n",
      "Found 82 species\n",
      "Found 71 transitions\n",
      "Found 118 interactions\n",
      "  Converted to spreadsheet in 0.075s\n",
      "Found 82 species\n",
      "Found 71 transitions\n",
      "Found 118 interactions\n",
      "  Converted back to SBML in 0.130s\n",
      "\n",
      "[157/205] Processing: 093_IMMUNE-CHECKPOINT-INHIBITORS_source.sbml\n",
      "Found 66 species\n",
      "Found 51 transitions\n",
      "Found 128 interactions\n",
      "  Converted to spreadsheet in 0.075s\n",
      "Found 66 species\n",
      "Found 51 transitions\n",
      "Found 128 interactions\n",
      "  Converted back to SBML in 0.102s\n",
      "\n",
      "[158/205] Processing: 041_INFLUENZA-VIRUS-REPLICATION-CYCLE_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[159/205] Processing: 119_JNK-PATHWAY_source.sbml\n",
      "Found 19 species\n",
      "Found 13 transitions\n",
      "Found 21 interactions\n",
      "  Converted to spreadsheet in 0.058s\n",
      "Found 19 species\n",
      "Found 13 transitions\n",
      "Found 21 interactions\n",
      "  Converted back to SBML in 0.089s\n",
      "\n",
      "[160/205] Processing: 067_REGULATION-OF-L-ARABINOSE-OPERON_source.sbml\n",
      "Found 13 species\n",
      "Found 9 transitions\n",
      "Found 18 interactions\n",
      "  Converted to spreadsheet in 0.054s\n",
      "Found 13 species\n",
      "Found 9 transitions\n",
      "Found 18 interactions\n",
      "  Converted back to SBML in 0.085s\n",
      "\n",
      "[161/205] Processing: 083_SIGNALING-IN-PROSTATE-CANCER_source.sbml\n",
      "Found 133 species\n",
      "Found 122 transitions\n",
      "Found 420 interactions\n",
      "  Converted to spreadsheet in 0.100s\n",
      "Found 133 species\n",
      "Found 122 transitions\n",
      "Found 420 interactions\n",
      "  Converted back to SBML in 0.178s\n",
      "\n",
      "[162/205] Processing: 153_CONTROL-OF-PROLIFERATION_source.sbml\n",
      "Found 18 species\n",
      "Found 17 transitions\n",
      "Found 34 interactions\n",
      "  Converted to spreadsheet in 0.062s\n",
      "Found 18 species\n",
      "Found 17 transitions\n",
      "Found 34 interactions\n",
      "  Converted back to SBML in 0.084s\n",
      "\n",
      "[163/205] Processing: 182_BOOLEAN-CELL-CYCLE_source.sbml\n",
      "Warning: Transition for target 'p27_1_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 14 species\n",
      "Found 13 transitions\n",
      "Found 59 interactions\n",
      "  Converted to spreadsheet in 0.053s\n",
      "Found 14 species\n",
      "Found 13 transitions\n",
      "Found 59 interactions\n",
      "  Converted back to SBML in 0.087s\n",
      "\n",
      "[164/205] Processing: 117_IFN-LAMBDA_source.sbml\n",
      "Found 47 species\n",
      "Found 28 transitions\n",
      "Found 52 interactions\n",
      "  Converted to spreadsheet in 0.068s\n",
      "Found 47 species\n",
      "Found 28 transitions\n",
      "Found 52 interactions\n",
      "  Converted back to SBML in 0.086s\n",
      "\n",
      "[165/205] Processing: 089_MAPK-REDUCED-1_source.sbml\n",
      "Found 17 species\n",
      "Found 13 transitions\n",
      "Found 78 interactions\n",
      "  Converted to spreadsheet in 0.055s\n",
      "Found 17 species\n",
      "Found 13 transitions\n",
      "Found 78 interactions\n",
      "  Converted back to SBML in 0.092s\n",
      "\n",
      "[166/205] Processing: 091_MAPK-REDUCED-3_source.sbml\n",
      "Found 16 species\n",
      "Found 12 transitions\n",
      "Found 58 interactions\n",
      "  Converted to spreadsheet in 0.050s\n",
      "Found 16 species\n",
      "Found 12 transitions\n",
      "Found 58 interactions\n",
      "  Converted back to SBML in 0.087s\n",
      "\n",
      "[167/205] Processing: 242_RHEUMATOID-ARTHRITIS-FIBROBLAST_source.sbml\n",
      "Found 371 species\n",
      "Found 286 transitions\n",
      "Found 622 interactions\n",
      "  Converted to spreadsheet in 0.509s\n",
      "Found 371 species\n",
      "Found 286 transitions\n",
      "Found 622 interactions\n",
      "  Converted back to SBML in 0.303s\n",
      "\n",
      "[168/205] Processing: 060_STOMATAL-OPENING_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "\n",
      "\n",
      "[169/205] Processing: 028_VEGF-PATHWAY-OF-DROSOPHILA_source.sbml\n",
      "Found 18 species\n",
      "Found 10 transitions\n",
      "Found 18 interactions\n",
      "  Converted to spreadsheet in 0.049s\n",
      "Found 18 species\n",
      "Found 10 transitions\n",
      "Found 18 interactions\n",
      "  Converted back to SBML in 0.076s\n",
      "\n",
      "[170/205] Processing: 048_GLUCOSE-REPRESSION-SIGNALING-2009_source.sbml\n",
      "Found 73 species\n",
      "Found 55 transitions\n",
      "Found 97 interactions\n",
      "  Converted to spreadsheet in 0.083s\n",
      "Found 73 species\n",
      "Found 55 transitions\n",
      "Found 97 interactions\n",
      "  Converted back to SBML in 0.109s\n",
      "\n",
      "[171/205] Processing: 197_ANTERIOR-POSTERIOR-BOUNDARY_source.sbml\n",
      "Warning: Transition for target 'CiR_1_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CiR_2_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CiR_3_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CiR_4_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'En_1_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'En_2_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'En_3_b2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'En_4_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 56 species\n",
      "Found 56 transitions\n",
      "Found 253 interactions\n",
      "  Converted to spreadsheet in 0.092s\n",
      "Found 56 species\n",
      "Found 56 transitions\n",
      "Found 253 interactions\n",
      "  Converted back to SBML in 0.201s\n",
      "\n",
      "[172/205] Processing: 080_TCR-SIGNALING-2018_source.sbml\n",
      "Warning: Transition for target 'CD4' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CD45' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CD6' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'RASA1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'GADD45A' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Malt1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CCBL' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'PTPN22' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'RCAN1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Akap5' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Bcl10' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'RAC1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Kinases' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 110 species\n",
      "Found 108 transitions\n",
      "Found 212 interactions\n",
      "  Converted to spreadsheet in 0.115s\n",
      "Warning: Species 'ABL1' (row 2): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'AICytokines' (row 3): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'AP1' (row 4): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'ARP' (row 5): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'ActinRem' (row 6): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Akap5' (row 7): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Akt' (row 8): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Anergy' (row 9): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'BAD' (row 10): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Bcl10' (row 11): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Bclxl' (row 12): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CALM1' (row 13): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CARD11' (row 14): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CARD11a' (row 15): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CBLB' (row 16): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CCBL' (row 17): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CCBLp' (row 18): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CD28' (row 19): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CD28p' (row 20): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CD4' (row 21): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CD45' (row 22): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CD6' (row 23): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CD6p' (row 24): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CDC42' (row 25): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CDKN1A' (row 26): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CDKN1B' (row 27): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CREB1' (row 28): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CSK' (row 29): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CTNNB1' (row 30): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Ca' (row 31): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Cabin1' (row 32): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Camk2' (row 33): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Camk4' (row 34): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'CellCycleProg' (row 35): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Cyc1' (row 36): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'DAG' (row 37): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'DGKZ' (row 38): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'DUSP' (row 39): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'ERK' (row 40): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'FOS' (row 41): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'FYN' (row 42): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Foxo1' (row 43): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'GAB2' (row 44): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'GADD45A' (row 45): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'GRAP2' (row 46): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'GRB2' (row 47): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'GSK3b' (row 48): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'HCLS1' (row 49): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'HRAS' (row 50): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'IKBKB' (row 51): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'IKBKG' (row 52): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'INPP5D' (row 53): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'IP3' (row 54): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'ITK' (row 55): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'JNK' (row 56): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'JUN' (row 57): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Kinases' (row 58): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'LAT' (row 59): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'LCK' (row 60): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'LCKr' (row 61): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'LCP2' (row 62): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'MAP2K1' (row 63): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'MAP2K4' (row 64): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'MAP3K1' (row 65): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'MAP3K11' (row 66): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'MAP3K8' (row 67): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'MAP4K1' (row 68): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'MAP4K3' (row 69): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Malt1' (row 70): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'NFAT' (row 71): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'NFKB1' (row 72): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'NFKBIA' (row 73): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'NFkB' (row 74): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'P38a' (row 75): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'PAG1' (row 76): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'PDPK1' (row 77): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'PI3K' (row 78): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'PICytokines' (row 79): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'PIK3R1' (row 80): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'PIP3' (row 81): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'PKCtheta' (row 82): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'PLCg' (row 83): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'PPP3C' (row 84): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'PTEN' (row 85): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'PTPN11' (row 86): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'PTPN22' (row 87): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'PTPN6' (row 88): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'RAC1' (row 89): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'RAC1p1' (row 90): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'RAC1p2' (row 91): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'RAF1' (row 92): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'RASA1' (row 93): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'RASGRP1' (row 94): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'RCAN1' (row 95): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'RHEBL1' (row 96): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'RPS6KA1' (row 97): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'SH3BP2' (row 98): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'SOS' (row 99): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'SRF' (row 100): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'Survival' (row 101): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'TCR' (row 102): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'TCRp' (row 103): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'TNFAIP3' (row 104): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'TXK' (row 105): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'VAV1' (row 106): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'VAV2' (row 107): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'WAS' (row 108): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'WAVE' (row 109): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'ZAP70' (row 110): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Warning: Species 'mTOR' (row 111): Invalid Relation 'unknownQualifier'. Valid values: is, hasVersion, isVersionOf, isDescribedBy, hasPart, isPartOf, hasProperty, isPropertyOf, encodes, isEncodedBy, isHomologTo, occursIn, hasTaxon\n",
      "Found 110 species\n",
      "Found 108 transitions\n",
      "Found 212 interactions\n",
      "  Converted back to SBML in 0.125s\n",
      "\n",
      "[173/205] Processing: 171_DROSOPHILA-GAP-C_source.sbml\n",
      "Warning: Transition for target 'Hb_b1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'Hb_b3' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 7 species\n",
      "Found 7 transitions\n",
      "Found 20 interactions\n",
      "  Converted to spreadsheet in 0.051s\n",
      "Found 7 species\n",
      "Found 7 transitions\n",
      "Found 20 interactions\n",
      "  Converted back to SBML in 0.078s\n",
      "\n",
      "[174/205] Processing: 050_CD4-T-CELL-SIGNALING_source.sbml\n",
      "Found 188 species\n",
      "Found 154 transitions\n",
      "Found 351 interactions\n",
      "  Converted to spreadsheet in 0.139s\n",
      "Found 188 species\n",
      "Found 154 transitions\n",
      "Found 351 interactions\n",
      "  Converted back to SBML in 0.144s\n",
      "\n",
      "[175/205] Processing: 158_LAMBDA-PHAGE-LYSOGENY_source.sbml\n",
      "Found 7 species\n",
      "Found 7 transitions\n",
      "Found 30 interactions\n",
      "  Converted to spreadsheet in 0.048s\n",
      "Found 7 species\n",
      "Found 7 transitions\n",
      "Found 30 interactions\n",
      "  Converted back to SBML in 0.076s\n",
      "\n",
      "[176/205] Processing: 204_HUMAN-BRAIN-ORGANOIDS_source.sbml\n",
      "Found 77 species\n",
      "Found 76 transitions\n",
      "Found 205 interactions\n",
      "  Converted to spreadsheet in 0.080s\n",
      "Found 77 species\n",
      "Found 76 transitions\n",
      "Found 205 interactions\n",
      "  Converted back to SBML in 0.108s\n",
      "\n",
      "[177/205] Processing: 112_COAGULATION-PATHWAY_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[178/205] Processing: 124_NSP9-PROTEIN_source.sbml\n",
      "Found 252 species\n",
      "Found 119 transitions\n",
      "Found 257 interactions\n",
      "  Converted to spreadsheet in 0.138s\n",
      "Found 252 species\n",
      "Found 119 transitions\n",
      "Found 257 interactions\n",
      "  Converted back to SBML in 0.149s\n",
      "\n",
      "[179/205] Processing: 154_CONTROL-OF-TH1-TH2-DIFFERENTIATION_source.sbml\n",
      "Warning: Transition for target 'IFNb' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL18' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL12' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 21 species\n",
      "Found 21 transitions\n",
      "Found 50 interactions\n",
      "  Converted to spreadsheet in 0.051s\n",
      "Found 21 species\n",
      "Found 21 transitions\n",
      "Found 50 interactions\n",
      "  Converted back to SBML in 0.088s\n",
      "\n",
      "[180/205] Processing: 168_DROSOPHILA-NOTCH-PATHWAY_source.sbml\n",
      "Found 13 species\n",
      "Found 7 transitions\n",
      "Found 26 interactions\n",
      "  Converted to spreadsheet in 0.052s\n",
      "Found 13 species\n",
      "Found 7 transitions\n",
      "Found 26 interactions\n",
      "  Converted back to SBML in 0.115s\n",
      "\n",
      "[181/205] Processing: 012_T-CELL-RECEPTOR-SIGNALING_source.sbml\n",
      "Found 101 species\n",
      "Found 94 transitions\n",
      "Found 158 interactions\n",
      "  Converted to spreadsheet in 0.101s\n",
      "Found 101 species\n",
      "Found 94 transitions\n",
      "Found 158 interactions\n",
      "  Converted back to SBML in 0.123s\n",
      "\n",
      "[182/205] Processing: 052_SEPTATION-INITIATION-NETWORK_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[183/205] Processing: 215_MTOR-PATHWAY_source.sbml\n",
      "Found 67 species\n",
      "Found 41 transitions\n",
      "Found 83 interactions\n",
      "  Converted to spreadsheet in 0.091s\n",
      "Found 67 species\n",
      "Found 41 transitions\n",
      "Found 83 interactions\n",
      "  Converted back to SBML in 0.109s\n",
      "\n",
      "[184/205] Processing: 026_BUDDING-YEAST-CELL-CYCLE-2009_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[185/205] Processing: 217_PRKN-PATHWAY_source.sbml\n",
      "Found 56 species\n",
      "Found 21 transitions\n",
      "Found 72 interactions\n",
      "  Converted to spreadsheet in 0.081s\n",
      "Found 56 species\n",
      "Found 21 transitions\n",
      "Found 72 interactions\n",
      "  Converted back to SBML in 0.103s\n",
      "\n",
      "[186/205] Processing: 114_ETC_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[187/205] Processing: 002_SIGNAL-TRANSDUCTION-IN-FIBROBLASTS_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[188/205] Processing: 127_PAMP-SIGNALING_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[189/205] Processing: 180_MORPHOGENETIC-CHECKPOINT_source.sbml\n",
      "Found 12 species\n",
      "Found 11 transitions\n",
      "Found 36 interactions\n",
      "  Converted to spreadsheet in 0.061s\n",
      "Found 12 species\n",
      "Found 11 transitions\n",
      "Found 36 interactions\n",
      "  Converted back to SBML in 0.098s\n",
      "\n",
      "[190/205] Processing: 051_COLITIS-ASSOCIATED-COLON-CANCER_source.sbml\n",
      "Found 70 species\n",
      "Found 69 transitions\n",
      "Found 153 interactions\n",
      "  Converted to spreadsheet in 0.101s\n",
      "Found 70 species\n",
      "Found 69 transitions\n",
      "Found 153 interactions\n",
      "  Converted back to SBML in 0.111s\n",
      "\n",
      "[191/205] Processing: 160_IL17-DIFFERENTIAL-EXPRESSION_source.sbml\n",
      "Warning: Transition for target 'IL12RB1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL12RB2' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL1RAP' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL1R1' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'IL2RB' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'CGC' has blank or empty rule. Setting to 1 (default level).\n",
      "Warning: Transition for target 'GP130' has blank or empty rule. Setting to 1 (default level).\n",
      "Found 92 species\n",
      "Found 83 transitions\n",
      "Found 246 interactions\n",
      "  Converted to spreadsheet in 0.081s\n",
      "Found 92 species\n",
      "Found 83 transitions\n",
      "Found 246 interactions\n",
      "  Converted back to SBML in 0.132s\n",
      "\n",
      "[192/205] Processing: 001_SIGNALING-IN-MACROPHAGE-ACTIVATION_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[193/205] Processing: 056_IGVH-MUTATIONS-IN-LYMPHOCYTIC-LEUKEMIA_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[194/205] Processing: 243_RHEUMATOID-ARTHRITIS-MULTI-CELLULAR_source.sbml\n",
      "  Error converting to spreadsheet: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "\n",
      "[195/205] Processing: 033_BT474-BREAST-CELL-LINE-LONG-TERM_source.sbml\n",
      "Found 24 species\n",
      "Found 19 transitions\n",
      "Found 68 interactions\n",
      "  Converted to spreadsheet in 0.056s\n",
      "Found 24 species\n",
      "Found 19 transitions\n",
      "Found 68 interactions\n",
      "  Converted back to SBML in 0.091s\n",
      "\n",
      "[196/205] Processing: 129_RTC-AND-TRANSCRIPTION_source.sbml\n",
      "Found 34 species\n",
      "Found 33 transitions\n",
      "Found 40 interactions\n",
      "  Converted to spreadsheet in 0.055s\n",
      "Found 34 species\n",
      "Found 33 transitions\n",
      "Found 40 interactions\n",
      "  Converted back to SBML in 0.083s\n",
      "\n",
      "[197/205] Processing: 192_SEGMENT-POLARITY-6-CELL_source.sbml\n",
      "Found 102 species\n",
      "Found 102 transitions\n",
      "Found 352 interactions\n",
      "  Converted to spreadsheet in 0.090s\n",
      "Found 102 species\n",
      "Found 102 transitions\n",
      "Found 352 interactions\n",
      "  Converted back to SBML in 0.129s\n",
      "\n",
      "[198/205] Processing: 030_SPZ-NETWORK-OF-DROSOPHILA_source.sbml\n",
      "Found 24 species\n",
      "Found 18 transitions\n",
      "Found 28 interactions\n",
      "  Converted to spreadsheet in 0.053s\n",
      "Found 24 species\n",
      "Found 18 transitions\n",
      "Found 28 interactions\n",
      "  Converted back to SBML in 0.079s\n",
      "\n",
      "[199/205] Processing: 024_BUDDING-YEAST-CELL-CYCLE_source.sbml\n",
      "Found 20 species\n",
      "Found 16 transitions\n",
      "Found 42 interactions\n",
      "  Converted to spreadsheet in 0.050s\n",
      "Found 20 species\n",
      "Found 16 transitions\n",
      "Found 42 interactions\n",
      "  Converted back to SBML in 0.081s\n",
      "\n",
      "[200/205] Processing: 285_MITOCHONDRIAL-DISFUNCTION_source.sbml\n",
      "Found 133 species\n",
      "Found 128 transitions\n",
      "Found 686 interactions\n",
      "  Converted to spreadsheet in 0.167s\n",
      "Found 133 species\n",
      "Found 128 transitions\n",
      "Found 686 interactions\n",
      "  Converted back to SBML in 0.974s\n",
      "\n",
      "[201/205] Processing: 074_T-LGL-SURVIVAL-NETWORK-2011-REDUCED_source.sbml\n",
      "Found 18 species\n",
      "Found 18 transitions\n",
      "Found 43 interactions\n",
      "  Converted to spreadsheet in 0.057s\n",
      "Found 18 species\n",
      "Found 18 transitions\n",
      "Found 43 interactions\n",
      "  Converted back to SBML in 0.083s\n",
      "\n",
      "[202/205] Processing: 027_WG-PATHWAY-OF-DROSOPHILA_source.sbml\n",
      "Found 26 species\n",
      "Found 12 transitions\n",
      "Found 29 interactions\n",
      "  Converted to spreadsheet in 0.064s\n",
      "Found 26 species\n",
      "Found 12 transitions\n",
      "Found 29 interactions\n",
      "  Converted back to SBML in 0.119s\n",
      "\n",
      "[203/205] Processing: 011_GUARD-CELL-ABSCISIC-ACID-SIGNALING_source.sbml\n",
      "Found 44 species\n",
      "Found 40 transitions\n",
      "Found 78 interactions\n",
      "  Converted to spreadsheet in 0.083s\n",
      "Found 44 species\n",
      "Found 40 transitions\n",
      "Found 78 interactions\n",
      "  Converted back to SBML in 0.105s\n",
      "\n",
      "[204/205] Processing: 275_MERGE-AML-PROGRESSION-100-273_source.sbml\n",
      "Found 50 species\n",
      "Found 45 transitions\n",
      "Found 0 interactions\n",
      "  Converted to spreadsheet in 0.055s\n",
      "Found 50 species\n",
      "Found 45 transitions\n",
      "Found 0 interactions\n",
      "  Converted back to SBML in 0.099s\n",
      "\n",
      "[205/205] Processing: 121_NLRP3-ACTIVATION_source.sbml\n",
      "Found 57 species\n",
      "Found 39 transitions\n",
      "Found 91 interactions\n",
      "  Converted to spreadsheet in 0.080s\n",
      "Found 57 species\n",
      "Found 39 transitions\n",
      "Found 91 interactions\n",
      "  Converted back to SBML in 0.101s\n",
      "\n",
      "\n",
      "Processed 205 models\n"
     ]
    }
   ],
   "source": [
    "import time\n",
    "\n",
    "results = []\n",
    "\n",
    "# Get all SBML files\n",
    "model_files = [f for f in os.listdir(biodivine_model_dir) \n",
    "               if (f.endswith('.xml') or f.endswith('.sbml')) and os.path.isfile(os.path.join(biodivine_model_dir, f))]\n",
    "print(f\"Found {len(model_files)} SBML models to process\")\n",
    "\n",
    "for idx, file_name in enumerate(model_files):\n",
    "    print(f\"\\n[{idx+1}/{len(model_files)}] Processing: {file_name}\")\n",
    "    \n",
    "    original_sbml_path = os.path.join(biodivine_model_dir, file_name)\n",
    "    base_name = os.path.splitext(file_name)[0]\n",
    "    \n",
    "    # Output paths\n",
    "    spreadsheet_path = os.path.join(spreadsheet_output_dir, f\"{base_name}.xlsx\")\n",
    "    converted_sbml_path = os.path.join(sbml_output_dir, f\"{base_name}_converted.sbml\")\n",
    "    \n",
    "    result = {\n",
    "        \"model\": file_name,\n",
    "        \"original_sbml_species\": None,\n",
    "        \"original_sbml_transitions\": None,\n",
    "        \"original_sbml_species_anno\": None,\n",
    "        \"original_sbml_trans_anno\": None,\n",
    "        \"spreadsheet_species\": None,\n",
    "        \"spreadsheet_transitions\": None,\n",
    "        \"spreadsheet_species_anno\": None,\n",
    "        \"spreadsheet_trans_anno\": None,\n",
    "        \"converted_sbml_species\": None,\n",
    "        \"converted_sbml_transitions\": None,\n",
    "        \"converted_sbml_species_anno\": None,\n",
    "        \"converted_sbml_trans_anno\": None,\n",
    "        \"sbml_to_xlsx_time\": None,\n",
    "        \"xlsx_to_sbml_time\": None,\n",
    "        \"sbml_to_xlsx_error\": None,\n",
    "        \"xlsx_to_sbml_error\": None,\n",
    "    }\n",
    "    \n",
    "    # Count stats from original SBML\n",
    "    orig_stats = count_sbml_stats(original_sbml_path)\n",
    "    if orig_stats.get(\"error\"):\n",
    "        result[\"sbml_to_xlsx_error\"] = orig_stats[\"error\"]\n",
    "        results.append(result)\n",
    "        print(f\"  Error reading original SBML: {orig_stats['error']}\")\n",
    "        continue\n",
    "    \n",
    "    result[\"original_sbml_species\"] = orig_stats[\"species_count\"]\n",
    "    result[\"original_sbml_transitions\"] = orig_stats[\"transitions_count\"]\n",
    "    result[\"original_sbml_species_anno\"] = orig_stats[\"species_annotations\"]\n",
    "    result[\"original_sbml_trans_anno\"] = orig_stats[\"transition_annotations\"]\n",
    "    \n",
    "    # Convert SBML to spreadsheet\n",
    "    try:\n",
    "        with warnings.catch_warnings():\n",
    "            warnings.simplefilter(\"ignore\")\n",
    "            start_time = time.time()\n",
    "            convert_sbml_to_spreadsheet(\n",
    "                original_sbml_path, \n",
    "                spreadsheet_path, \n",
    "                template_path=None,  # No template to skip README/Appendix\n",
    "                rule_format=\"operators\"\n",
    "            )\n",
    "            sbml_to_xlsx_time = time.time() - start_time\n",
    "            result[\"sbml_to_xlsx_time\"] = round(sbml_to_xlsx_time, 3)\n",
    "        print(f\"  Converted to spreadsheet in {sbml_to_xlsx_time:.3f}s\")\n",
    "    except Exception as e:\n",
    "        result[\"sbml_to_xlsx_error\"] = str(e)\n",
    "        results.append(result)\n",
    "        print(f\"  Error converting to spreadsheet: {e}\")\n",
    "        continue\n",
    "    \n",
    "    # Count stats from spreadsheet\n",
    "    xlsx_stats = count_spreadsheet_stats(spreadsheet_path)\n",
    "    if xlsx_stats.get(\"error\"):\n",
    "        result[\"sbml_to_xlsx_error\"] = xlsx_stats[\"error\"]\n",
    "    else:\n",
    "        result[\"spreadsheet_species\"] = xlsx_stats[\"species_count\"]\n",
    "        result[\"spreadsheet_transitions\"] = xlsx_stats[\"transitions_count\"]\n",
    "        result[\"spreadsheet_species_anno\"] = xlsx_stats[\"species_annotations\"]\n",
    "        result[\"spreadsheet_trans_anno\"] = xlsx_stats[\"transition_annotations\"]\n",
    "    \n",
    "    # Convert spreadsheet back to SBML\n",
    "    try:\n",
    "        with warnings.catch_warnings():\n",
    "            warnings.simplefilter(\"ignore\")\n",
    "            start_time = time.time()\n",
    "            convert_spreadsheet_to_sbml(\n",
    "                spreadsheet_path,\n",
    "                converted_sbml_path,\n",
    "                interactions_anno=False, # no existing interaction annotations\n",
    "                transitions_anno=True\n",
    "            )\n",
    "            xlsx_to_sbml_time = time.time() - start_time\n",
    "            result[\"xlsx_to_sbml_time\"] = round(xlsx_to_sbml_time, 3)\n",
    "        print(f\"  Converted back to SBML in {xlsx_to_sbml_time:.3f}s\")\n",
    "    except Exception as e:\n",
    "        result[\"xlsx_to_sbml_error\"] = str(e)\n",
    "        results.append(result)\n",
    "        print(f\"  Error converting to SBML: {e}\")\n",
    "        continue\n",
    "    \n",
    "    # Count stats from converted SBML\n",
    "    conv_stats = count_sbml_stats(converted_sbml_path)\n",
    "    if conv_stats.get(\"error\"):\n",
    "        result[\"xlsx_to_sbml_error\"] = conv_stats[\"error\"]\n",
    "    else:\n",
    "        result[\"converted_sbml_species\"] = conv_stats[\"species_count\"]\n",
    "        result[\"converted_sbml_transitions\"] = conv_stats[\"transitions_count\"]\n",
    "        result[\"converted_sbml_species_anno\"] = conv_stats[\"species_annotations\"]\n",
    "        result[\"converted_sbml_trans_anno\"] = conv_stats[\"transition_annotations\"]\n",
    "    \n",
    "    results.append(result)\n",
    "\n",
    "print(f\"\\n\\nProcessed {len(results)} models\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Results saved to: biodivine_260125_results.csv\n",
      "\n",
      "=== Summary ===\n",
      "Total models: 205\n",
      "Successfully converted: 172\n",
      "Species count matches: 172/172\n",
      "Transitions count matches: 172/172\n",
      "\n",
      "Models with errors (33):\n",
      "  054_PC12-CELL-DIFFERENTIATION_source.sbml: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "\n",
      "  126_ORF3A_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  078_IMMUNE-SYSTEM_source.sbml: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  072_LYMPHOPOIESIS-REGULATORY-NETWORK_source.sbml: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "\n",
      "  066_CD4-T-CELL-DIFFERENTIATION_source.sbml: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  218_TCA-CYCLE_source.sbml: SBML parsing errors: The value of the attribute 'qual:sign' of an <input> object must conform to the syntax of the SBML data type 'sign' and may only take on the allowed values of 'sign' defined in SBML; that is, the value must be one of the following: 'positive', 'negative', 'dual' or 'unknown'.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      " The inputSign on the <input> with id 'tr_sa11967_in_0' is 'mixed', which is not a valid option.\n",
      "\n",
      "  047_FGF-PATHWAY-OF-DROSOPHILA_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  077_SIGNALLING-PATHWAY-FOR-BUTANOL-PRODUCTION_source.sbml: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  145_MELANOGENESIS_source.sbml: SBML parsing errors: A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; A <qualitativeSpecies> object must have the required attributes 'qual:id', 'qual:compartment' and 'qual:constant', and may have the optional attributes 'qual:name', 'qual:initialLevel' and 'qual:maxLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on a <qualitativeSpecies> object. \n",
      "Reference: L3V1 Qual V1 Section 3.5\n",
      " Qual attribute 'compartment' is missing.\n",
      "; An SBML XML document must conform to the XML Schema for the corresponding SBML Level, Version and Release. The XML Schema for SBML defines the basic SBML object structure, the data types used by those objects, and the order in which the objects may appear in an SBML document.\n",
      " listOfInputs cannot be empty.\n",
      "\n",
      "  122_NSP14_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  159_BUDDING-YEAST-CORE_source.sbml: Unexpected end of expression. Missing closing parenthesis ')'. Found 294 opening '(' but only 293 closing ')' in: (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & Cln2 & !Swi5 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & !Swi5 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & !Swi5 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & !Swi5 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & !Swi5 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & !Swi5 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1\n",
      "  003_MAMMALIAN-CELL-CYCLE_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  071_CASTRATION-RESISTANT-PROSTATE-CANCER_source.sbml: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "\n",
      "  005_FA-BRCA-PATHWAY_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  059_BORTEZOMIB-RESPONSES-IN-MYELOMA-CELLS_source.sbml: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  272_FLT3-ITD-AML-SIGNALING_source.sbml: SBML parsing errors: An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object must have the required attributes 'qual:qualitativeSpecies' as well as 'qual:transitionEffect' and may have the optional attributes 'qual:id', 'qual:name', 'qual:sign' and 'qual:thresholdLevel'.  No other attributes from the SBML Level 3 Qualitative Models namespace are permitted on an <input> object. \n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  004_ERBB-RECEPTOR-SIGNALING_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  064_METABOLIC-INTERACTIONS-IN-GUT-MICROBIOME_source.sbml: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "\n",
      "  057_FANCONI-ANEMIA-AND-CHECKPOINT-RECOVERY_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  146_BUDDING-YEAST-FAURE-2009_source.sbml: Unexpected end of expression. Missing closing parenthesis ')'. Found 277 opening '(' but only 276 closing ')' in: (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & Cln2 & !Swi5 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & !Swi5 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & !Swi5 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & !Bck2_b1 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & !Clb5_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & !Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & !Swi5 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & !Swi5 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & !Swi5 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & !Clb2_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Clb5_b3 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & !Bck2_b2 & Cln2 & Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Clb5_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & !Sic1 & Cdc6 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & !Swi5 & Sic1 & Clb5_b1 & Clb5_b2 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & !Clb5_b1 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & !Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & !Clb2_b1) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & !Clb2_b2) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & !Clb2_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & !Net1_b1 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & !Net1_b2 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & !Cdc6 & Clb5_b1 & Clb5_b2 & Clb2_b1 & Clb2_b2 & Clb2_b3 & Net1_b1 & Net1_b2 & !Net1_b3 & Cdc14_b1 & Cdc14_b2 & Cdc14_b3) | (!Cln3 & Bck2_b1 & Bck2_b2 & !Cln2 & Swi5 & !Sic1 & Cdc6 & !Clb5_b1) | (!Cln3 & Bck2_b1\n",
      "  055_HUMAN-GONADAL-SEX-DETERMINATION_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  039_HIV-1-INTERACTIONS-WITH-T-CELL-SIGNALING_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  041_INFLUENZA-VIRUS-REPLICATION-CYCLE_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  060_STOMATAL-OPENING_source.sbml: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "\n",
      "  112_COAGULATION-PATHWAY_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  052_SEPTATION-INITIATION-NETWORK_source.sbml: SBML parsing errors: A <layout> object must have the required attribute 'layout:id' and may have the optional attribute 'layout:name'. No other attributes from the SBML Level 3 Layout namespace are permitted on a <layout> object.\n",
      "Reference: L3V1 Layout V1 Section 3.6\n",
      " Layout attribute 'id' is missing.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  026_BUDDING-YEAST-CELL-CYCLE-2009_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  114_ETC_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  002_SIGNAL-TRANSDUCTION-IN-FIBROBLASTS_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  127_PAMP-SIGNALING_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  001_SIGNALING-IN-MACROPHAGE-ACTIVATION_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  056_IGVH-MUTATIONS-IN-LYMPHOCYTIC-LEUKEMIA_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n",
      "  243_RHEUMATOID-ARTHRITIS-MULTI-CELLULAR_source.sbml: SBML parsing errors: An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "; An <input> object may have the optional 'metaid' and 'sboTerm' defined by SBML Level 3 Core.  No other attributes from the SBML Level 3 Core namespace or the Qualitative Models namespace are permitted on an <input> object.\n",
      "Reference: L3V1 Qual V1 Section 3.6.1\n",
      "  Attribute 'essential' is not part of the definition of an SBML Level 3 Version 1 Package qual Version 1 <input> element.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Create results DataFrame and save to CSV\n",
    "results_df = pd.DataFrame(results)\n",
    "\n",
    "# Add comparison columns to check if counts match\n",
    "results_df[\"species_match\"] = (\n",
    "    (results_df[\"original_sbml_species\"] == results_df[\"spreadsheet_species\"]) & \n",
    "    (results_df[\"spreadsheet_species\"] == results_df[\"converted_sbml_species\"])\n",
    ")\n",
    "results_df[\"transitions_match\"] = (\n",
    "    (results_df[\"original_sbml_transitions\"] == results_df[\"spreadsheet_transitions\"]) & \n",
    "    (results_df[\"spreadsheet_transitions\"] == results_df[\"converted_sbml_transitions\"])\n",
    ")\n",
    "# For annotations, we compare original to spreadsheet (converted may differ due to URI format)\n",
    "results_df[\"species_anno_match\"] = (\n",
    "    results_df[\"original_sbml_species_anno\"] == results_df[\"spreadsheet_species_anno\"]\n",
    ")\n",
    "results_df[\"trans_anno_match\"] = (\n",
    "    results_df[\"original_sbml_trans_anno\"] == results_df[\"spreadsheet_trans_anno\"]\n",
    ")\n",
    "\n",
    "# Reorder columns for better readability\n",
    "column_order = [\n",
    "    \"model\",\n",
    "    \"original_sbml_species\", \"spreadsheet_species\", \"converted_sbml_species\", \"species_match\",\n",
    "    \"original_sbml_transitions\", \"spreadsheet_transitions\", \"converted_sbml_transitions\", \"transitions_match\",\n",
    "    \"original_sbml_species_anno\", \"spreadsheet_species_anno\", \"converted_sbml_species_anno\", \"species_anno_match\",\n",
    "    \"original_sbml_trans_anno\", \"spreadsheet_trans_anno\", \"converted_sbml_trans_anno\", \"trans_anno_match\",\n",
    "    \"sbml_to_xlsx_time\", \"xlsx_to_sbml_time\",\n",
    "    \"sbml_to_xlsx_error\", \"xlsx_to_sbml_error\"\n",
    "]\n",
    "results_df = results_df[[c for c in column_order if c in results_df.columns]]\n",
    "\n",
    "# Save to CSV\n",
    "output_csv = \"biodivine_260125_results.csv\"\n",
    "results_df.to_csv(output_csv, index=False)\n",
    "print(f\"Results saved to: {output_csv}\")\n",
    "\n",
    "# Summary statistics\n",
    "total = len(results_df)\n",
    "successful = results_df[results_df[\"sbml_to_xlsx_error\"].isna() & results_df[\"xlsx_to_sbml_error\"].isna()]\n",
    "species_match = successful[\"species_match\"].sum()\n",
    "trans_match = successful[\"transitions_match\"].sum()\n",
    "\n",
    "print(f\"\\n=== Summary ===\")\n",
    "print(f\"Total models: {total}\")\n",
    "print(f\"Successfully converted: {len(successful)}\")\n",
    "print(f\"Species count matches: {species_match}/{len(successful)}\")\n",
    "print(f\"Transitions count matches: {trans_match}/{len(successful)}\")\n",
    "\n",
    "# Show errors if any\n",
    "errors = results_df[results_df[\"sbml_to_xlsx_error\"].notna() | results_df[\"xlsx_to_sbml_error\"].notna()]\n",
    "if len(errors) > 0:\n",
    "    print(f\"\\nModels with errors ({len(errors)}):\")\n",
    "    for _, row in errors.iterrows():\n",
    "        print(f\"  {row['model']}: {row['sbml_to_xlsx_error'] or row['xlsx_to_sbml_error']}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "=== Annotation Statistics ===\n",
      "\n",
      "Original SBML annotations:\n",
      "  Species annotations: 34693\n",
      "  Transition annotations: 2924\n",
      "\n",
      "Spreadsheet annotations:\n",
      "  Species annotations: 34736.0\n",
      "  Transition annotations: 2924.0\n",
      "\n",
      "Converted SBML annotations:\n",
      "  Species annotations: 25919.0\n",
      "  Transition annotations: 2649.0\n",
      "\n",
      "Models with annotation count differences (5):\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>model</th>\n",
       "      <th>original_sbml_species_anno</th>\n",
       "      <th>spreadsheet_species_anno</th>\n",
       "      <th>converted_sbml_species_anno</th>\n",
       "      <th>original_sbml_trans_anno</th>\n",
       "      <th>spreadsheet_trans_anno</th>\n",
       "      <th>converted_sbml_trans_anno</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>241_RHEUMATOID-ARTHRITIS-TH1_source.sbml</td>\n",
       "      <td>674</td>\n",
       "      <td>676.0</td>\n",
       "      <td>527.0</td>\n",
       "      <td>71</td>\n",
       "      <td>71.0</td>\n",
       "      <td>68.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>130_RENIN-ANGIOTENSIN_source.sbml</td>\n",
       "      <td>238</td>\n",
       "      <td>246.0</td>\n",
       "      <td>241.0</td>\n",
       "      <td>208</td>\n",
       "      <td>208.0</td>\n",
       "      <td>173.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>113_ER-STRESS_source.sbml</td>\n",
       "      <td>631</td>\n",
       "      <td>634.0</td>\n",
       "      <td>634.0</td>\n",
       "      <td>361</td>\n",
       "      <td>361.0</td>\n",
       "      <td>302.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>242_RHEUMATOID-ARTHRITIS-FIBROBLAST_source.sbml</td>\n",
       "      <td>17086</td>\n",
       "      <td>17112.0</td>\n",
       "      <td>9553.0</td>\n",
       "      <td>76</td>\n",
       "      <td>76.0</td>\n",
       "      <td>76.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>124_NSP9-PROTEIN_source.sbml</td>\n",
       "      <td>788</td>\n",
       "      <td>792.0</td>\n",
       "      <td>788.0</td>\n",
       "      <td>300</td>\n",
       "      <td>300.0</td>\n",
       "      <td>299.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               model  \\\n",
       "61          241_RHEUMATOID-ARTHRITIS-TH1_source.sbml   \n",
       "102                130_RENIN-ANGIOTENSIN_source.sbml   \n",
       "150                        113_ER-STRESS_source.sbml   \n",
       "166  242_RHEUMATOID-ARTHRITIS-FIBROBLAST_source.sbml   \n",
       "177                     124_NSP9-PROTEIN_source.sbml   \n",
       "\n",
       "     original_sbml_species_anno  spreadsheet_species_anno  \\\n",
       "61                          674                     676.0   \n",
       "102                         238                     246.0   \n",
       "150                         631                     634.0   \n",
       "166                       17086                   17112.0   \n",
       "177                         788                     792.0   \n",
       "\n",
       "     converted_sbml_species_anno  original_sbml_trans_anno  \\\n",
       "61                         527.0                        71   \n",
       "102                        241.0                       208   \n",
       "150                        634.0                       361   \n",
       "166                       9553.0                        76   \n",
       "177                        788.0                       300   \n",
       "\n",
       "     spreadsheet_trans_anno  converted_sbml_trans_anno  \n",
       "61                     71.0                       68.0  \n",
       "102                   208.0                      173.0  \n",
       "150                   361.0                      302.0  \n",
       "166                    76.0                       76.0  \n",
       "177                   300.0                      299.0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "=== Timing Statistics ===\n",
      "Total SBML â†’ Spreadsheet time: 13.07s\n",
      "Total Spreadsheet â†’ SBML time: 20.23s\n",
      "Average SBML â†’ Spreadsheet time: 0.076s\n",
      "Average Spreadsheet â†’ SBML time: 0.118s\n",
      "Max SBML â†’ Spreadsheet time: 0.509s\n",
      "Max Spreadsheet â†’ SBML time: 0.974s\n"
     ]
    }
   ],
   "source": [
    "# Annotation analysis\n",
    "# Note: Original SBML often uses \"urn:miriam\" URIs while the converter \n",
    "# produces identifiers.org URLs. The counts should still match.\n",
    "\n",
    "print(\"=== Annotation Statistics ===\")\n",
    "print(f\"\\nOriginal SBML annotations:\")\n",
    "print(f\"  Species annotations: {successful['original_sbml_species_anno'].sum()}\")\n",
    "print(f\"  Transition annotations: {successful['original_sbml_trans_anno'].sum()}\")\n",
    "\n",
    "print(f\"\\nSpreadsheet annotations:\")\n",
    "print(f\"  Species annotations: {successful['spreadsheet_species_anno'].sum()}\")\n",
    "print(f\"  Transition annotations: {successful['spreadsheet_trans_anno'].sum()}\")\n",
    "\n",
    "print(f\"\\nConverted SBML annotations:\")\n",
    "print(f\"  Species annotations: {successful['converted_sbml_species_anno'].sum()}\")\n",
    "print(f\"  Transition annotations: {successful['converted_sbml_trans_anno'].sum()}\")\n",
    "\n",
    "# Show models with annotation differences\n",
    "anno_diff = successful[\n",
    "    (successful['original_sbml_species_anno'] != successful['spreadsheet_species_anno']) |\n",
    "    (successful['original_sbml_trans_anno'] != successful['spreadsheet_trans_anno'])\n",
    "]\n",
    "if len(anno_diff) > 0:\n",
    "    print(f\"\\nModels with annotation count differences ({len(anno_diff)}):\")\n",
    "    anno_cols = [\"model\", \n",
    "                 \"original_sbml_species_anno\", \"spreadsheet_species_anno\", \"converted_sbml_species_anno\",\n",
    "                 \"original_sbml_trans_anno\", \"spreadsheet_trans_anno\", \"converted_sbml_trans_anno\"]\n",
    "    display(anno_diff[anno_cols].head(20))\n",
    "else:\n",
    "    print(\"\\nAll annotation counts match between original SBML and spreadsheet!\")\n",
    "\n",
    "# Timing statistics\n",
    "print(f\"\\n=== Timing Statistics ===\")\n",
    "print(f\"Total SBML â†’ Spreadsheet time: {successful['sbml_to_xlsx_time'].sum():.2f}s\")\n",
    "print(f\"Total Spreadsheet â†’ SBML time: {successful['xlsx_to_sbml_time'].sum():.2f}s\")\n",
    "print(f\"Average SBML â†’ Spreadsheet time: {successful['sbml_to_xlsx_time'].mean():.3f}s\")\n",
    "print(f\"Average Spreadsheet â†’ SBML time: {successful['xlsx_to_sbml_time'].mean():.3f}s\")\n",
    "print(f\"Max SBML â†’ Spreadsheet time: {successful['sbml_to_xlsx_time'].max():.3f}s\")\n",
    "print(f\"Max Spreadsheet â†’ SBML time: {successful['xlsx_to_sbml_time'].max():.3f}s\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mismatching annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "# For models with annotation count differences, extract and compare annotations\n",
    "\n",
    "def get_sbml_species_annotations(sbml_path):\n",
    "    \"\"\"Extract all species annotations from SBML file as a dict of {species_id: [uri, ...]}\"\"\"\n",
    "    reader = libsbml.SBMLReader()\n",
    "    document = reader.readSBML(sbml_path)\n",
    "    model = document.getModel()\n",
    "    if model is None:\n",
    "        return {}\n",
    "    \n",
    "    qual_plugin = model.getPlugin(\"qual\")\n",
    "    if not qual_plugin:\n",
    "        return {}\n",
    "    \n",
    "    annotations = {}\n",
    "    for i in range(qual_plugin.getNumQualitativeSpecies()):\n",
    "        qs = qual_plugin.getQualitativeSpecies(i)\n",
    "        species_id = qs.getId()\n",
    "        if qs.isSetAnnotation():\n",
    "            anno_str = qs.getAnnotationString()\n",
    "            # Extract all resource URIs\n",
    "            resources = re.findall(r'<rdf:li[^>]*resource=\"([^\"]+)\"', anno_str)\n",
    "            annotations[species_id] = resources\n",
    "        else:\n",
    "            annotations[species_id] = []\n",
    "    return annotations\n",
    "\n",
    "\n",
    "def get_spreadsheet_species_annotations(xlsx_path):\n",
    "    \"\"\"Extract all species annotations from spreadsheet as a dict of {species_id: [uri, ...]}\"\"\"\n",
    "    from TabularQual_converter.spreadsheet_reader import read_spreadsheet_to_model\n",
    "    model, _ = read_spreadsheet_to_model(xlsx_path)\n",
    "    \n",
    "    annotations = {}\n",
    "    for species_id, sp in model.species.items():\n",
    "        # Annotations are stored as (qualifier, identifier) tuples\n",
    "        annotations[species_id] = [ident for qual, ident in sp.annotations]\n",
    "    return annotations\n",
    "\n",
    "\n",
    "def normalize_uri(uri):\n",
    "    \"\"\"Normalize URI to comparable format (strip urn:miriam: prefix, convert to identifiers.org style)\"\"\"\n",
    "    # # Handle urn:miriam format: urn:miriam:TYPE:ID -> TYPE:ID\n",
    "    # if uri.startswith(\"urn:miriam:\"):\n",
    "    #     parts = uri[11:].split(\":\", 1)\n",
    "    #     if len(parts) == 2:\n",
    "    #         return f\"{parts[0]}:{parts[1]}\"\n",
    "    #     return uri[11:]\n",
    "    # # Handle identifiers.org format: https://identifiers.org/TYPE/ID -> TYPE:ID\n",
    "    # if \"identifiers.org/\" in uri:\n",
    "    #     match = re.search(r'identifiers\\.org/([^/]+)/([^/]+)$', uri)\n",
    "    #     if match:\n",
    "    #         return f\"{match.group(1)}:{match.group(2)}\"\n",
    "    # # If already in TYPE:ID form (e.g., pubmed:12345), just return as is\n",
    "    # match_simple = re.match(r'^[^:/]+:[^:/]+$', uri)\n",
    "    # if match_simple:\n",
    "    #     return uri\n",
    "\n",
    "    # Just remove the prefix \"https://identifiers.org/\" or \"urn:miriam:\"\n",
    "    return uri.replace(\"https://identifiers.org/\", \"\").replace(\"urn:miriam:\", \"\")\n",
    "\n",
    "\n",
    "def compare_annotations(model_file):\n",
    "    base_name = os.path.splitext(model_file)[0]\n",
    "    original_sbml_path = os.path.join(biodivine_model_dir, model_file)\n",
    "    spreadsheet_path = os.path.join(spreadsheet_output_dir, f\"{base_name}.xlsx\")\n",
    "    converted_sbml_path = os.path.join(sbml_output_dir, f\"{base_name}_converted.sbml\")\n",
    "    \n",
    "    # Get annotations from each source\n",
    "    with warnings.catch_warnings():\n",
    "        warnings.simplefilter(\"ignore\")\n",
    "        orig_annos = get_sbml_species_annotations(original_sbml_path)\n",
    "        xlsx_annos = get_spreadsheet_species_annotations(spreadsheet_path)\n",
    "        conv_annos = get_sbml_species_annotations(converted_sbml_path)\n",
    "    \n",
    "    # Compare original SBML vs Spreadsheet\n",
    "    print(f\"\\n--- SBML â†’ Spreadsheet differences ---\")\n",
    "    orig_total = sum(len(v) for v in orig_annos.values())\n",
    "    xlsx_total = sum(len(v) for v in xlsx_annos.values())\n",
    "    print(f\"Original SBML: {orig_total} annotations, Spreadsheet: {xlsx_total} annotations\")\n",
    "    \n",
    "    for species_id in set(orig_annos.keys()) | set(xlsx_annos.keys()):\n",
    "        orig_set = set(normalize_uri(u) for u in orig_annos.get(species_id, []))\n",
    "        xlsx_set = set(normalize_uri(u) for u in xlsx_annos.get(species_id, []))\n",
    "        \n",
    "        only_in_orig = orig_set - xlsx_set\n",
    "        only_in_xlsx = xlsx_set - orig_set\n",
    "        \n",
    "        if only_in_orig or only_in_xlsx:\n",
    "            print(f\"\\n  Species '{species_id}':\")\n",
    "            if only_in_orig:\n",
    "                print(f\"    Only in original SBML ({len(only_in_orig)}):\")\n",
    "                for u in list(only_in_orig)[:10]:  # Show max 10\n",
    "                    print(f\"      - {u}\")\n",
    "                if len(only_in_orig) > 10:\n",
    "                    print(f\"      ... and {len(only_in_orig) - 10} more\")\n",
    "            if only_in_xlsx:\n",
    "                print(f\"    Only in spreadsheet ({len(only_in_xlsx)}):\")\n",
    "                for u in list(only_in_xlsx)[:10]:\n",
    "                    print(f\"      - {u}\")\n",
    "                if len(only_in_xlsx) > 10:\n",
    "                    print(f\"      ... and {len(only_in_xlsx) - 10} more\")\n",
    "    \n",
    "    # Compare Spreadsheet vs Converted SBML\n",
    "    print(f\"\\n--- Spreadsheet â†’ SBML differences ---\")\n",
    "    conv_total = sum(len(v) for v in conv_annos.values())\n",
    "    print(f\"Spreadsheet: {xlsx_total} annotations, Converted SBML: {conv_total} annotations\")\n",
    "    \n",
    "    diff_count = 0\n",
    "    for species_id in set(xlsx_annos.keys()) | set(conv_annos.keys()):\n",
    "        xlsx_set = set(normalize_uri(u) for u in xlsx_annos.get(species_id, []))\n",
    "        conv_set = set(normalize_uri(u) for u in conv_annos.get(species_id, []))\n",
    "        \n",
    "        only_in_xlsx = xlsx_set - conv_set\n",
    "        only_in_conv = conv_set - xlsx_set\n",
    "        \n",
    "        if only_in_xlsx or only_in_conv:\n",
    "            diff_count += 1\n",
    "            if diff_count <= 10:  # Show details for first 10 species only\n",
    "                print(f\"\\n  Species '{species_id}':\")\n",
    "                if only_in_xlsx:\n",
    "                    print(f\"    Only in spreadsheet ({len(only_in_xlsx)}):\")\n",
    "                    for u in list(only_in_xlsx)[:10]:\n",
    "                        print(f\"      - {u}\")\n",
    "                    if len(only_in_xlsx) > 10:\n",
    "                        print(f\"      ... and {len(only_in_xlsx) - 10} more\")\n",
    "                if only_in_conv:\n",
    "                    print(f\"    Only in converted SBML ({len(only_in_conv)}):\")\n",
    "                    for u in list(only_in_conv)[:10]:\n",
    "                        print(f\"      - {u}\")\n",
    "    \n",
    "    if diff_count > 10:\n",
    "        print(f\"\\n  ... and {diff_count - 10} more species with differences\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "--- SBML â†’ Spreadsheet differences ---\n",
      "Original SBML: 674 annotations, Spreadsheet: 676 annotations\n",
      "\n",
      "  Species 'sa3388':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:(CCL4,5/CCR5_complex|CXCL10/CXCR3_complex|AGT/F2R_complex|CCL2/CCR2_complex)&amp;GNAI/GNB/GNG_complex\n",
      "    Only in spreadsheet (2):\n",
      "      - urn:casq:function:(CCL4\n",
      "      - 5/CCR5_complex|CXCL10/CXCR3_complex|AGT/F2R_complex|CCL2/CCR2_complex)&GNAI/GNB/GNG_complex\n",
      "\n",
      "  Species 'sa3423':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:NFKB1/RELA/NFKBIA_complex&amp;IKK_complex\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:NFKB1/RELA/NFKBIA_complex&IKK_complex\n",
      "\n",
      "  Species 'sa3537':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:VAV1&amp;!RHOA&amp;!SEMA4/PLXNB1_complex\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:VAV1&!RHOA&!SEMA4/PLXNB1_complex\n",
      "\n",
      "  Species 'sa2925':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:STAT4/STAT4_complex&amp;tbx21_phosphorylated&amp;RUNX3&amp;EOMES\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:STAT4/STAT4_complex&tbx21_phosphorylated&RUNX3&EOMES\n",
      "\n",
      "  Species 'sa3859':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:LTA_TH1 : secreted components&amp;!IL2\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:LTA_TH1 : secreted components&!IL2\n",
      "\n",
      "  Species 'csa421':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:IL18_TH1 : extracellular space&amp;IL18R1/IL18RAP_complex\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:IL18_TH1 : extracellular space&IL18R1/IL18RAP_complex\n",
      "\n",
      "  Species 'csa424':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:LGALS9&amp;!PDCD1\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:LGALS9&!PDCD1\n",
      "\n",
      "  Species 'sa2982':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:NICD/EP300/SKIP/CSL/MAML1_complex&amp;!RUNX3&amp;!tbx21_phosphorylated\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:NICD/EP300/SKIP/CSL/MAML1_complex&!RUNX3&!tbx21_phosphorylated\n",
      "\n",
      "  Species 'csa343':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:IL12_TH1 : extracellular space&amp;IL12Rb1/IL12Rb2_complex\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:IL12_TH1 : extracellular space&IL12Rb1/IL12Rb2_complex\n",
      "\n",
      "  Species 'csa411':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:(CCL4,5/CCR5_complex|CXCL10/CXCR3_complex|AGT/F2R_complex|CCL2/CCR2_complex)&amp;GNAI/GNB/GNG_complex\n",
      "    Only in spreadsheet (2):\n",
      "      - urn:casq:function:(CCL4\n",
      "      - 5/CCR5_complex|CXCL10/CXCR3_complex|AGT/F2R_complex|CCL2/CCR2_complex)&GNAI/GNB/GNG_complex\n",
      "\n",
      "  Species 'sa3735':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:IRF7_TH1 : cytoplasm&amp;TBK1\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:IRF7_TH1 : cytoplasm&TBK1\n",
      "\n",
      "  Species 'sa2945':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:STAT4/STAT4_complex&amp;STAT1/STAT1_complex\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:STAT4/STAT4_complex&STAT1/STAT1_complex\n",
      "\n",
      "  Species 'sa3766':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:HLA-DP/DQ/DR/TCR/CD3_complex&amp;!LGALS9/TIM3_complex&amp;!HLA-DP/DQ/DR/LAG3_complex\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:HLA-DP/DQ/DR/TCR/CD3_complex&!LGALS9/TIM3_complex&!HLA-DP/DQ/DR/LAG3_complex\n",
      "\n",
      "  Species 'csa409':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:(CCL4_TH1 : extracellular space&amp;CCR5|CCL5_TH1 : extracellular space&amp;CCR5)\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:(CCL4_TH1 : extracellular space&CCR5|CCL5_TH1 : extracellular space&CCR5)\n",
      "\n",
      "  Species 'csa415':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:NFKB1/RELA/NFKBIA_complex&amp;IKK_complex\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:NFKB1/RELA/NFKBIA_complex&IKK_complex\n",
      "\n",
      "  Species 'csa441':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:EP300&amp;NICD&amp;SKIP&amp;CSL&amp;MAML1\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:EP300&NICD&SKIP&CSL&MAML1\n",
      "\n",
      "  Species 'sa2929':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:STAT4/STAT4_complex&amp;NFKB1/RELA_complex_TH1 : nucleus&amp;POU2F1\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:STAT4/STAT4_complex&NFKB1/RELA_complex_TH1 : nucleus&POU2F1\n",
      "\n",
      "  Species 'sa2941':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:tbx21&amp;mtor\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:tbx21&mtor\n",
      "\n",
      "  Species 'csa442':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:(SEMA4D&amp;PLXNB1|SEMA4A_TH1 : extracellular space&amp;PLXNB1)\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:(SEMA4D&PLXNB1|SEMA4A_TH1 : extracellular space&PLXNB1)\n",
      "\n",
      "  Species 'sa3481':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:GNAI_TH1 : cytoplasm&amp;CXCL16/CXCR6_complex\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:GNAI_TH1 : cytoplasm&CXCL16/CXCR6_complex\n",
      "\n",
      "  Species 'sa2930':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:(LTA_TH1 : secreted components|LGALS9/TIM3_complex|BCL2L11)&amp;!NFKB1/RELA_complex_TH1 : cytoplasm\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:(LTA_TH1 : secreted components|LGALS9/TIM3_complex|BCL2L11)&!NFKB1/RELA_complex_TH1 : cytoplasm\n",
      "\n",
      "  Species 'csa407':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:IFNg_TH1 : extracellular space&amp;IFNGR1/IFNGR2_complex\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:IFNg_TH1 : extracellular space&IFNGR1/IFNGR2_complex\n",
      "\n",
      "  Species 'sa3408':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:(Src|GNB/GNG_complex|GNAI_TH1 : cytoplasm_active|CD28/CD86_complex|Rac1|gas6/mertk_complex|PTK2|icoslg/icos_complex|TNFSF4/TNFRSF4_complex|JAK1/JAK3_complex|TRAF1/TRAF2/TRAF3_complex)&amp;!LGALS9/TIM3_complex&amp;!SHIP1\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:(Src|GNB/GNG_complex|GNAI_TH1 : cytoplasm_active|CD28/CD86_complex|Rac1|gas6/mertk_complex|PTK2|icoslg/icos_complex|TNFSF4/TNFRSF4_complex|JAK1/JAK3_complex|TRAF1/TRAF2/TRAF3_complex)&!LGALS9/TIM3_complex&!SHIP1\n",
      "\n",
      "  Species 'sa3417':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:(MAPK1_phosphorylated|CREB1_phosphorylated|IL12_TH1 : secreted components|NFKB1/RELA_complex_TH1 : cytoplasm)&amp;!LGALS9/TIM3_complex&amp;!BCL2L11\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:(MAPK1_phosphorylated|CREB1_phosphorylated|IL12_TH1 : secreted components|NFKB1/RELA_complex_TH1 : cytoplasm)&!LGALS9/TIM3_complex&!BCL2L11\n",
      "\n",
      "--- Spreadsheet â†’ SBML differences ---\n",
      "Spreadsheet: 676 annotations, Converted SBML: 527 annotations\n",
      "\n",
      "  Species 'sa3388':\n",
      "    Only in spreadsheet (1):\n",
      "      - 5/CCR5_complex|CXCL10/CXCR3_complex|AGT/F2R_complex|CCL2/CCR2_complex)&GNAI/GNB/GNG_complex\n",
      "    Only in converted SBML (1):\n",
      "      - 5:CCR5_complex|CXCL10/CXCR3_complex|AGT/F2R_complex|CCL2/CCR2_complex)&amp;GNAI/GNB/GNG_complex\n",
      "\n",
      "  Species 'sa3423':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:NFKB1/RELA/NFKBIA_complex&IKK_complex\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:NFKB1/RELA/NFKBIA_complex&amp;IKK_complex\n",
      "\n",
      "  Species 'sa3537':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:VAV1&!RHOA&!SEMA4/PLXNB1_complex\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:VAV1&amp;!RHOA&amp;!SEMA4/PLXNB1_complex\n",
      "\n",
      "  Species 'sa2925':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:STAT4/STAT4_complex&tbx21_phosphorylated&RUNX3&EOMES\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:STAT4/STAT4_complex&amp;tbx21_phosphorylated&amp;RUNX3&amp;EOMES\n",
      "\n",
      "  Species 'sa3859':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:LTA_TH1 : secreted components&!IL2\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:LTA_TH1 : secreted components&amp;!IL2\n",
      "\n",
      "  Species 'csa421':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:IL18_TH1 : extracellular space&IL18R1/IL18RAP_complex\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:IL18_TH1 : extracellular space&amp;IL18R1/IL18RAP_complex\n",
      "\n",
      "  Species 'csa424':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:LGALS9&!PDCD1\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:LGALS9&amp;!PDCD1\n",
      "\n",
      "  Species 'sa2982':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:NICD/EP300/SKIP/CSL/MAML1_complex&!RUNX3&!tbx21_phosphorylated\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:NICD/EP300/SKIP/CSL/MAML1_complex&amp;!RUNX3&amp;!tbx21_phosphorylated\n",
      "\n",
      "  Species 'csa343':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:IL12_TH1 : extracellular space&IL12Rb1/IL12Rb2_complex\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:IL12_TH1 : extracellular space&amp;IL12Rb1/IL12Rb2_complex\n",
      "\n",
      "  Species 'csa411':\n",
      "    Only in spreadsheet (1):\n",
      "      - 5/CCR5_complex|CXCL10/CXCR3_complex|AGT/F2R_complex|CCL2/CCR2_complex)&GNAI/GNB/GNG_complex\n",
      "    Only in converted SBML (1):\n",
      "      - 5:CCR5_complex|CXCL10/CXCR3_complex|AGT/F2R_complex|CCL2/CCR2_complex)&amp;GNAI/GNB/GNG_complex\n",
      "\n",
      "  ... and 14 more species with differences\n"
     ]
    }
   ],
   "source": [
    "compare_annotations(\"241_RHEUMATOID-ARTHRITIS-TH1_source.sbml\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "--- SBML â†’ Spreadsheet differences ---\n",
      "Original SBML: 238 annotations, Spreadsheet: 246 annotations\n",
      "\n",
      "  Species 'sa23':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:angiotensin I_simple_molecule&amp;ACE_cell|S|angiotensin 1-12_simple_molecule&amp;CMA1|angiotensin I_simple_molecule&amp;CMA1|angiotensin I_simple_molecule&amp;KLK1\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:angiotensin I_simple_molecule&ACE_cell|S|angiotensin 1-12_simple_molecule&CMA1|angiotensin I_simple_molecule&CMA1|angiotensin I_simple_molecule&KLK1\n",
      "\n",
      "  Species 'csa4':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:AGTR1_cell&amp;ACE2_cell&amp;!angiotensin II_simple_molecule_cell\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:AGTR1_cell&ACE2_cell&!angiotensin II_simple_molecule_cell\n",
      "\n",
      "  Species 'sa50':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:MRGPRD_cell&amp;alamandine_simple_molecule|MRGPRD_cell&amp;angiotensin 1-7_simple_molecule_cell\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:MRGPRD_cell&alamandine_simple_molecule|MRGPRD_cell&angiotensin 1-7_simple_molecule_cell\n",
      "\n",
      "  Species 'sa22':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:angiotensin I_simple_molecule&amp;ACE2_cell\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:angiotensin I_simple_molecule&ACE2_cell\n",
      "\n",
      "  Species 'csa9':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:AGTR1_cell_active&amp;MAS1_cell\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:AGTR1_cell_active&MAS1_cell\n",
      "\n",
      "  Species 'sa25':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:AGTR2_cell&amp;CGP42112A_drug|angiotensin II_simple_molecule_cell|angiotensin A_simple_molecule&amp;AGTR2_cell|AGTR2_cell&amp;angiotensin 1-7_simple_molecule_cell|AGTR2_cell&amp;angiotensin 1-9_simple_molecule\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:AGTR2_cell&CGP42112A_drug|angiotensin II_simple_molecule_cell|angiotensin A_simple_molecule&AGTR2_cell|AGTR2_cell&angiotensin 1-7_simple_molecule_cell|AGTR2_cell&angiotensin 1-9_simple_molecule\n",
      "\n",
      "  Species 'sa28':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:estradiol_simple_molecule&amp;!Calcitriol_simple_molecule\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:estradiol_simple_molecule&!Calcitriol_simple_molecule\n",
      "\n",
      "  Species 'sa63':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:angiotensin III_simple_molecule&amp;ANPEP \n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:angiotensin III_simple_molecule&ANPEP\n",
      "\n",
      "  Species 'sa156':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:LNPEP_cell&amp;angiotensin IV_simple_molecule|LNPEP_cell&amp;angiotensin 3-7_simple_molecule\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:LNPEP_cell&angiotensin IV_simple_molecule|LNPEP_cell&angiotensin 3-7_simple_molecule\n",
      "\n",
      "  Species 'sa102':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:angiotensin II_simple_molecule_cell|angiotensin IV_simple_molecule|angiotensin A_simple_molecule&amp;AGTR1_cell|AGTR1_cell&amp;!Losartan_drug|AGTR1_cell&amp;!angiotensin 1-7_simple_molecule_cell|angiotensin II_simple_molecule_cell&amp;!MAS1:AGTR1_complex\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:angiotensin II_simple_molecule_cell|angiotensin IV_simple_molecule|angiotensin A_simple_molecule&AGTR1_cell|AGTR1_cell&!Losartan_drug|AGTR1_cell&!angiotensin 1-7_simple_molecule_cell|angiotensin II_simple_molecule_cell&!MAS1:AGTR1_complex\n",
      "\n",
      "  Species 'sa44':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:sex, male_phenotype\n",
      "    Only in spreadsheet (2):\n",
      "      - urn:casq:function:sex\n",
      "      - male_phenotype\n",
      "\n",
      "  Species 'sa49':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:angiotensin 1-7_simple_molecule_cell|angiotensin A_simple_molecule&amp;ACE2_cell\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:angiotensin 1-7_simple_molecule_cell|angiotensin A_simple_molecule&ACE2_cell\n",
      "\n",
      "  Species 'sa65':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:ANPEP \n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:ANPEP\n",
      "\n",
      "  Species 'sa62':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:angiotensin II_simple_molecule_cell&amp;ENPEP_cell_active\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:angiotensin II_simple_molecule_cell&ENPEP_cell_active\n",
      "\n",
      "  Species 'sa194':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:MME_cell&amp;sex, male_phenotype\n",
      "    Only in spreadsheet (2):\n",
      "      - urn:casq:function:MME_cell&sex\n",
      "      - male_phenotype\n",
      "\n",
      "  Species 'sa64':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:ENPEP_cell&amp;!QGC001_drug|ENPEP_cell&amp;!sex, male_phenotype\n",
      "    Only in spreadsheet (2):\n",
      "      - urn:casq:function:ENPEP_cell&!QGC001_drug|ENPEP_cell&!sex\n",
      "      - male_phenotype\n",
      "\n",
      "  Species 'sa21':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:AGT&amp;REN|angiotensin 1-12_simple_molecule&amp;ACE_cell|CTSD|CTSG&amp;AGT\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:AGT&REN|angiotensin 1-12_simple_molecule&ACE_cell|CTSD|CTSG&AGT\n",
      "\n",
      "  Species 'sa30':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:estradiol_simple_molecule|Calcitriol_simple_molecule&amp;ACE2_gene&amp;!S|!SARS-CoV-2 infection_phenotype|ACE2_gene&amp;aging_phenotype|ACE2_gene&amp;!hypertension_phenotype&amp;! Diabetes mellitus, type II_phenotype\n",
      "    Only in spreadsheet (2):\n",
      "      - type II_phenotype\n",
      "      - urn:casq:function:estradiol_simple_molecule|Calcitriol_simple_molecule&ACE2_gene&!S|!SARS-CoV-2 infection_phenotype|ACE2_gene&aging_phenotype|ACE2_gene&!hypertension_phenotype&! Diabetes mellitus\n",
      "\n",
      "  Species 'sa203':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function: Diabetes mellitus, type II_phenotype\n",
      "    Only in spreadsheet (2):\n",
      "      - type II_phenotype\n",
      "      - urn:casq:function: Diabetes mellitus\n",
      "\n",
      "  Species 'sa27':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:MAS1_cell&amp;angiotensin 1-5_simple_molecule|MAS1_cell&amp;AR234960_drug|MAS1_cell&amp;angiotensin 1-7_simple_molecule_cell\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:MAS1_cell&angiotensin 1-5_simple_molecule|MAS1_cell&AR234960_drug|MAS1_cell&angiotensin 1-7_simple_molecule_cell\n",
      "\n",
      "  Species 'sa138':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:angiotensin 1-12_simple_molecule&amp;MME_cell\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:angiotensin 1-12_simple_molecule&MME_cell\n",
      "\n",
      "  Species 'sa96':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:angiotensin II_simple_molecule_human host&amp;ACE2, soluble\n",
      "    Only in spreadsheet (2):\n",
      "      - urn:casq:function:angiotensin II_simple_molecule_human host&ACE2\n",
      "      - soluble\n",
      "\n",
      "  Species 'sa73':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:ADAM17|AGTR1_cell_active&amp;ACE2_cell\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:ADAM17|AGTR1_cell_active&ACE2_cell\n",
      "\n",
      "  Species 'sa32':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:angiotensin 1-7_simple_molecule_cell&amp;ACE_cell_active\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:angiotensin 1-7_simple_molecule_cell&ACE_cell_active\n",
      "\n",
      "  Species 'sa100':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:ACE_cell&amp;!aging_phenotype&amp;!sex, male_phenotype|ACE_cell&amp;aldosterone_simple_molecule|ACE_cell&amp;hypertension_phenotype|ACE_cell&amp;!Lisinopril_drug|ACE_cell&amp;ABO blood group system_phenotype\n",
      "    Only in spreadsheet (2):\n",
      "      - male_phenotype|ACE_cell&aldosterone_simple_molecule|ACE_cell&hypertension_phenotype|ACE_cell&!Lisinopril_drug|ACE_cell&ABO blood group system_phenotype\n",
      "      - urn:casq:function:ACE_cell&!aging_phenotype&!sex\n",
      "\n",
      "  Species 'csa5':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:ACE2_cell&amp;S|S&amp;TMPRSS2&amp;!Camostat mesilate_drug\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:ACE2_cell&S|S&TMPRSS2&!Camostat mesilate_drug\n",
      "\n",
      "  Species 'sa24':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:ACE_cell_active|MME_cell&amp;angiotensin 1-9_simple_molecule|ACE2_cell|PRCP&amp;angiotensin II_simple_molecule_cell|angiotensin I_simple_molecule&amp;MME_cell|angiotensin 1-12_simple_molecule&amp;MME_cell|angiotensin I_simple_molecule&amp;THOP1|angiotensin I_simple_molecule&amp;PREP\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:ACE_cell_active|MME_cell&angiotensin 1-9_simple_molecule|ACE2_cell|PRCP&angiotensin II_simple_molecule_cell|angiotensin I_simple_molecule&MME_cell|angiotensin 1-12_simple_molecule&MME_cell|angiotensin I_simple_molecule&THOP1|angiotensin I_simple_molecule&PREP\n",
      "\n",
      "  Species 'sa98':\n",
      "    Only in original SBML (1):\n",
      "      - urn:casq:function:ACE2_cell&amp;sex, male_phenotype|aging_phenotype|hypertension_phenotype&amp;ACE2_cell\n",
      "    Only in spreadsheet (2):\n",
      "      - urn:casq:function:ACE2_cell&sex\n",
      "      - male_phenotype|aging_phenotype|hypertension_phenotype&ACE2_cell\n",
      "\n",
      "--- Spreadsheet â†’ SBML differences ---\n",
      "Spreadsheet: 246 annotations, Converted SBML: 241 annotations\n",
      "\n",
      "  Species 'sa23':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:angiotensin I_simple_molecule&ACE_cell|S|angiotensin 1-12_simple_molecule&CMA1|angiotensin I_simple_molecule&CMA1|angiotensin I_simple_molecule&KLK1\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:angiotensin I_simple_molecule&amp;ACE_cell|S|angiotensin 1-12_simple_molecule&amp;CMA1|angiotensin I_simple_molecule&amp;CMA1|angiotensin I_simple_molecule&amp;KLK1\n",
      "\n",
      "  Species 'csa4':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:AGTR1_cell&ACE2_cell&!angiotensin II_simple_molecule_cell\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:AGTR1_cell&amp;ACE2_cell&amp;!angiotensin II_simple_molecule_cell\n",
      "\n",
      "  Species 'sa50':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:MRGPRD_cell&alamandine_simple_molecule|MRGPRD_cell&angiotensin 1-7_simple_molecule_cell\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:MRGPRD_cell&amp;alamandine_simple_molecule|MRGPRD_cell&amp;angiotensin 1-7_simple_molecule_cell\n",
      "\n",
      "  Species 'sa22':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:angiotensin I_simple_molecule&ACE2_cell\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:angiotensin I_simple_molecule&amp;ACE2_cell\n",
      "\n",
      "  Species 'csa9':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:AGTR1_cell_active&MAS1_cell\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:AGTR1_cell_active&amp;MAS1_cell\n",
      "\n",
      "  Species 'sa25':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:AGTR2_cell&CGP42112A_drug|angiotensin II_simple_molecule_cell|angiotensin A_simple_molecule&AGTR2_cell|AGTR2_cell&angiotensin 1-7_simple_molecule_cell|AGTR2_cell&angiotensin 1-9_simple_molecule\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:AGTR2_cell&amp;CGP42112A_drug|angiotensin II_simple_molecule_cell|angiotensin A_simple_molecule&amp;AGTR2_cell|AGTR2_cell&amp;angiotensin 1-7_simple_molecule_cell|AGTR2_cell&amp;angiotensin 1-9_simple_molecule\n",
      "\n",
      "  Species 'sa28':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:estradiol_simple_molecule&!Calcitriol_simple_molecule\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:estradiol_simple_molecule&amp;!Calcitriol_simple_molecule\n",
      "\n",
      "  Species 'sa63':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:angiotensin III_simple_molecule&ANPEP\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:angiotensin III_simple_molecule&amp;ANPEP\n",
      "\n",
      "  Species 'sa156':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:LNPEP_cell&angiotensin IV_simple_molecule|LNPEP_cell&angiotensin 3-7_simple_molecule\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:LNPEP_cell&amp;angiotensin IV_simple_molecule|LNPEP_cell&amp;angiotensin 3-7_simple_molecule\n",
      "\n",
      "  Species 'sa102':\n",
      "    Only in spreadsheet (1):\n",
      "      - urn:casq:function:angiotensin II_simple_molecule_cell|angiotensin IV_simple_molecule|angiotensin A_simple_molecule&AGTR1_cell|AGTR1_cell&!Losartan_drug|AGTR1_cell&!angiotensin 1-7_simple_molecule_cell|angiotensin II_simple_molecule_cell&!MAS1:AGTR1_complex\n",
      "    Only in converted SBML (1):\n",
      "      - urn:casq:function:angiotensin II_simple_molecule_cell|angiotensin IV_simple_molecule|angiotensin A_simple_molecule&amp;AGTR1_cell|AGTR1_cell&amp;!Losartan_drug|AGTR1_cell&amp;!angiotensin 1-7_simple_molecule_cell|angiotensin II_simple_molecule_cell&amp;!MAS1:AGTR1_complex\n",
      "\n",
      "  ... and 15 more species with differences\n"
     ]
    }
   ],
   "source": [
    "compare_annotations(\"130_RENIN-ANGIOTENSIN_source.sbml\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "qual",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
